<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Accessing EuPathDB Resources using AnnotationHub</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Accessing EuPathDB Resources using AnnotationHub</h1>



<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script>
<p><strong>Authors</strong>: <a href="mailto:keith.hughitt@nih.gov">V. Keith Hughitt</a><br /> <strong>Modified:</strong> 2019-04-30 14:37:07<br /> <strong>Compiled</strong>: Tue Apr 30 14:32:43 2019</p>
<div id="overview" class="section level1">
<h1>Overview</h1>
<p>This tutorial describes how to query and make use of annotations retrieved from <a href="http://eupathdb.org/eupathdb/">EuPathDB : The Eukaryotic Pathogen Genomics Resource</a> using <a href="http://bioconductor.org/packages/release/bioc/html/AnnotationHub.html">AnnotationHub</a>.</p>
<p>For more information on using AnnotationHub, check out the AnnotationHub vignettes:</p>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/AnnotationHub/inst/doc/AnnotationHub-HOWTO.html">AnnotationHub: Access the AnnotationHub Web Service</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/AnnotationHub/inst/doc/AnnotationHub-HOWTO.html">AnnotationHub How-To’s</a></li>
</ul>
<p>The resources described in this tutorial were generating using GFF files and web API requests made to the various EuPathDB databases (TriTrypDB, ToxoDB, etc.) Only organisms with annotated genomes (those for which GFF files are available) are accessible through AnnotationHub.</p>
<p>The two main resources provided are:</p>
<ul>
<li><a href="https://www.bioconductor.org/help/workflows/annotation/annotation/#OrgDb">OrgDb</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">GRanges</a></li>
</ul>
<p>OrgDb objects for an organism include basic gene-level information such as:</p>
<ul>
<li>Gene ID</li>
<li>Gene description</li>
<li>Chromosome number</li>
<li>GO terms associated with gene</li>
<li>KEGG Pathways associated with gene</li>
<li>Etc.</li>
</ul>
<p>For some organisms, <a href="https://www.ebi.ac.uk/interpro/">InterPro</a> protein domain information is also available (in some cases, however, even though InterPro domain information is available through EuPathDB, it is too large to be included in the current AnnotationHub resources).</p>
<p>For more information about working with Bioconductor annotation resources, see:</p>
<ul>
<li><a href="https://www.bioconductor.org/help/workflows/annotation/annotation/">Genomic Annotation Resources in Bioconductor</a></li>
</ul>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>If you don’t already have AnnotationHub installed on your system, use <code>BiocManager::install</code> to install the package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</a>
<a class="sourceLine" id="cb1-2" title="2">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;AnnotationHub&quot;</span>)</a></code></pre></div>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<p>To begin, let’s create a new <code>AnnotationHub</code> connection and use it to query AnnotationHub for all EuPathDB resources.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(<span class="st">&quot;EuPathDB&quot;</span>)</a></code></pre></div>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colMeans, colnames, colSums, dirname, do.call, duplicated,
##     eval, evalq, Filter, Find, get, grep, grepl, intersect,
##     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,
##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,
##     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation(&quot;Biobase&quot;)', and for packages 'citation(&quot;pkgname&quot;)'.</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: GenomeInfoDbData</code></pre>
<pre><code>## Loading required package: AnnotationHub</code></pre>
<pre><code>## 
## Attaching package: 'AnnotationHub'</code></pre>
<pre><code>## The following object is masked from 'package:Biobase':
## 
##     cache</code></pre>
<pre><code>## Registered S3 method overwritten by 'rvest':
##   method            from
##   read_xml.response xml2</code></pre>
<pre><code>## 
## This is EuPathDB version 1.0.1
##  Read 'EuPathDB()' to get started.</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" title="1"><span class="kw">library</span>(<span class="st">&quot;AnnotationHub&quot;</span>)</a>
<a class="sourceLine" id="cb24-2" title="2"></a>
<a class="sourceLine" id="cb24-3" title="3"><span class="co"># create an AnnotationHub connection</span></a>
<a class="sourceLine" id="cb24-4" title="4">ah &lt;-<span class="st"> </span><span class="kw">AnnotationHub</span>()</a></code></pre></div>
<pre><code>## snapshotDate(): 2018-10-24</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" title="1"><span class="co"># search for all EuPathDB resources</span></a>
<a class="sourceLine" id="cb26-2" title="2">meta &lt;-<span class="st"> </span><span class="kw">query</span>(ah, <span class="st">&quot;EuPathDB&quot;</span>)</a>
<a class="sourceLine" id="cb26-3" title="3"></a>
<a class="sourceLine" id="cb26-4" title="4"><span class="kw">length</span>(meta)</a></code></pre></div>
<pre><code>## [1] 530</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" title="1"><span class="kw">head</span>(meta)</a></code></pre></div>
<pre><code>## AnnotationHub with 6 records
## # snapshotDate(): 2018-10-24 
## # $dataprovider: FungiDB, TriTrypDB, MicrosporidiaDB
## # $species: Aspergillus sydowii, Aspergillus versicolor, Edhazardia aed...
## # $rdataclass: OrgDb
## # additional mcols(): taxonomyid, genome, description,
## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass,
## #   tags, rdatapath, sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[[&quot;AH65001&quot;]]' 
## 
##             title                                                         
##   AH65001 | Edhazardia aedis USNM 41457 genome wide annotations           
##   AH65002 | Aspergillus versicolor CBS 583.65 genome wide annotations     
##   AH65003 | Aspergillus sydowii CBS 593.65 genome wide annotations        
##   AH65004 | Phytophthora cinnamomi var. cinnamomi CBS 144.22 genome wid...
##   AH65005 | Trypanosoma vivax Y486 genome wide annotations                
##   AH65006 | Trypanosoma congolense IL3000 genome wide annotations</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" title="1"><span class="co"># types of EuPathDB data available</span></a>
<a class="sourceLine" id="cb30-2" title="2"><span class="kw">table</span>(meta<span class="op">$</span>rdataclass)</a></code></pre></div>
<pre><code>## 
## GRanges   OrgDb 
##     265     265</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1"><span class="co"># distribution of resources by specific databases</span></a>
<a class="sourceLine" id="cb32-2" title="2"><span class="kw">table</span>(meta<span class="op">$</span>dataprovider)</a></code></pre></div>
<pre><code>## 
##        AmoebaDB        CryptoDB         FungiDB       GiardiaDB 
##              22              26             228               2 
## MicrosporidiaDB    PiroplasmaDB        PlasmoDB          ToxoDB 
##              60              18              54              52 
##         TrichDB       TriTrypDB 
##               2              66</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" title="1"><span class="co"># list of organisms for which resources are available</span></a>
<a class="sourceLine" id="cb34-2" title="2"><span class="kw">length</span>(<span class="kw">unique</span>(meta<span class="op">$</span>species))</a></code></pre></div>
<pre><code>## [1] 196</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" title="1"><span class="kw">head</span>(<span class="kw">unique</span>(meta<span class="op">$</span>species))</a></code></pre></div>
<pre><code>## [1] &quot;Edhazardia aedis&quot;       &quot;Aspergillus versicolor&quot;
## [3] &quot;Aspergillus sydowii&quot;    &quot;Phytophthora cinnamomi&quot;
## [5] &quot;Trypanosoma vivax&quot;      &quot;Trypanosoma congolense&quot;</code></pre>
</div>
<div id="working-with-eupathdb-orgdb-resources" class="section level1">
<h1>Working with EuPathDB OrgDb resources</h1>
<p>Next, we will see how you can query AnnotationHub for EuPathDB OrgDb resources.</p>
<p>To begin, create an AnnotationHub connection, if you have not already done so, as shown in the section above.</p>
<p>You can now use the <code>query</code> function to search for your organism of interest and store the result as follows:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" title="1">res &lt;-<span class="st"> </span><span class="kw">query</span>(ah, <span class="kw">c</span>(<span class="st">'Leishmania major strain Friedlin'</span>, <span class="st">'OrgDb'</span>, <span class="st">'EuPathDB'</span>))</a>
<a class="sourceLine" id="cb38-2" title="2">res</a></code></pre></div>
<pre><code>## AnnotationHub with 1 record
## # snapshotDate(): 2018-10-24 
## # names(): AH65089
## # $dataprovider: TriTrypDB
## # $species: Leishmania major
## # $rdataclass: OrgDb
## # $rdatadateadded: 2018-10-15
## # $title: Leishmania major strain Friedlin genome wide annotations
## # $description: TriTrypDB 39 genome annotations for Leishmania major st...
## # $taxonomyid: 347515
## # $genome: TriTrypDB-39_LmajorFriedlin
## # $sourcetype: GFF
## # $sourceurl: http://TriTrypDB.org/common/downloads/Current_Release/Lma...
## # $sourcesize: NA
## # $tags:
## #   c(&quot;Annotation,EuPathDB,Eukaryote,Pathogen,Parasite,Trypanosome,Kinetoplastid,Leishmania&quot;,
## #   &quot;EuPathDB&quot;) 
## # retrieve record with 'object[[&quot;AH65089&quot;]]'</code></pre>
<p>The result includes a single record, “AH56967”. The record can be accessed from the result variable using list-like indexing:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" title="1">orgdb_uid &lt;-<span class="st"> </span><span class="kw">names</span>(res<span class="op">@</span>.db_uid)</a>
<a class="sourceLine" id="cb40-2" title="2">orgdb &lt;-<span class="st"> </span>res[[orgdb_uid]]</a></code></pre></div>
<pre><code>## downloading 0 resources</code></pre>
<pre><code>## loading from cache 
##     '/home/trey//.AnnotationHub/71835'</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" title="1"><span class="kw">class</span>(orgdb)</a></code></pre></div>
<pre><code>## [1] &quot;OrgDb&quot;
## attr(,&quot;package&quot;)
## [1] &quot;AnnotationDbi&quot;</code></pre>
<p>We can see that we now have an OrgDb instance, and as such, we can use the usual methods available for working this OrgDb objects, including:</p>
<ul>
<li><code>columns()</code></li>
<li><code>keys()</code></li>
<li><code>select()</code></li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" title="1"><span class="co"># list available fields to retrieve</span></a>
<a class="sourceLine" id="cb46-2" title="2"><span class="kw">columns</span>(orgdb)</a></code></pre></div>
<pre><code>##  [1] &quot;CHR&quot;                   &quot;EVIDENCE_CODE&quot;        
##  [3] &quot;GENEDESCRIPTION&quot;       &quot;GENESOURCE&quot;           
##  [5] &quot;GENETYPE&quot;              &quot;GID&quot;                  
##  [7] &quot;GO_ID&quot;                 &quot;GO_TERM_NAME&quot;         
##  [9] &quot;INTERPRO_DESC&quot;         &quot;INTERPRO_E_VALUE&quot;     
## [11] &quot;INTERPRO_END_MIN&quot;      &quot;INTERPRO_FAMILY_ID&quot;   
## [13] &quot;INTERPRO_NAME&quot;         &quot;INTERPRO_PRIMARY_ID&quot;  
## [15] &quot;INTERPRO_SECONDARY_ID&quot; &quot;INTERPRO_START_MIN&quot;   
## [17] &quot;ONTOLOGY&quot;              &quot;PATHWAY&quot;              
## [19] &quot;PATHWAY_SOURCE&quot;        &quot;PATHWAY_SOURCE_ID&quot;    
## [21] &quot;SOURCE&quot;                &quot;TRANSCRIPT_IDS&quot;       
## [23] &quot;TYPE&quot;</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" title="1"><span class="co"># create a vector containing all gene ids for the organism</span></a>
<a class="sourceLine" id="cb48-2" title="2">gids &lt;-<span class="st"> </span><span class="kw">keys</span>(orgdb, <span class="dt">keytype=</span><span class="st">'GID'</span>)</a>
<a class="sourceLine" id="cb48-3" title="3"><span class="kw">head</span>(gids)</a></code></pre></div>
<pre><code>## [1] &quot;LmjF.24.0370&quot; &quot;LmjF.34.2180&quot; &quot;LmjF.36.2890&quot; &quot;LmjF.28.0420&quot;
## [5] &quot;LmjF.16.0980&quot; &quot;LmjF.22.0730&quot;</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1"><span class="co"># retrieve the chromosome, description, and biotype for each gene</span></a>
<a class="sourceLine" id="cb50-2" title="2">dat &lt;-<span class="st"> </span><span class="kw">select</span>(orgdb, <span class="dt">keys=</span>gids, <span class="dt">keytype=</span><span class="st">'GID'</span>, <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">'CHR'</span>, <span class="st">'TYPE'</span>, <span class="st">'GENEDESCRIPTION'</span>))</a></code></pre></div>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1"><span class="kw">head</span>(dat)</a></code></pre></div>
<pre><code>##            GID     CHR           TYPE
## 1 LmjF.24.0370 LmjF.24 protein coding
## 2 LmjF.34.2180 LmjF.34 protein coding
## 3 LmjF.36.2890 LmjF.36 protein coding
## 4 LmjF.28.0420 LmjF.28 protein coding
## 5 LmjF.16.0980 LmjF.16 protein coding
## 6 LmjF.22.0730 LmjF.22 protein coding
##                                                GENEDESCRIPTION
## 1                         aspartate aminotransferase, putative
## 2                              hypothetical protein, conserved
## 3 ATP-binding cassette protein subfamily G, member 6, putative
## 4                 hypothetical protein, conserved (pseudogene)
## 5                              hypothetical protein, conserved
## 6                    cytoskeleton associated protein, putative</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" title="1"><span class="kw">table</span>(dat<span class="op">$</span>TYPE)</a></code></pre></div>
<pre><code>## 
## non protein coding     protein coding      rRNA encoding 
##                 83               8402                 63 
##    snoRNA encoding     snRNA encoding      tRNA encoding 
##                741                  6                 83</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" title="1"><span class="kw">table</span>(dat<span class="op">$</span>CHR)</a></code></pre></div>
<pre><code>## 
## LmjF.01 LmjF.02 LmjF.03 LmjF.04 LmjF.05 LmjF.06 LmjF.07 LmjF.08 LmjF.09 
##      85     136      97     130     159     135     130     136     182 
## LmjF.10 LmjF.11 LmjF.12 LmjF.13 LmjF.14 LmjF.15 LmjF.16 LmjF.17 LmjF.18 
##     150     148     142     170     159     192     177     162     219 
## LmjF.19 LmjF.20 LmjF.21 LmjF.22 LmjF.23 LmjF.24 LmjF.25 LmjF.26 LmjF.27 
##     179     276     233     181     229     246     322     495     385 
## LmjF.28 LmjF.29 LmjF.30 LmjF.31 LmjF.32 LmjF.33 LmjF.34 LmjF.35 LmjF.36 
##     319     300     408     346     430     424     500     615     781</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" title="1"><span class="co"># create a gene / GO term mapping</span></a>
<a class="sourceLine" id="cb58-2" title="2">gene_go_mapping &lt;-<span class="st"> </span><span class="kw">select</span>(orgdb, <span class="dt">keys=</span>gids, <span class="dt">keytype=</span><span class="st">'GID'</span>,</a>
<a class="sourceLine" id="cb58-3" title="3">                          <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">'GO_ID'</span>, <span class="st">'GO_TERM_NAME'</span>, <span class="st">'ONTOLOGY'</span>))</a></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" title="1"><span class="kw">head</span>(gene_go_mapping)</a></code></pre></div>
<pre><code>##            GID      GO_ID                GO_TERM_NAME           ONTOLOGY
## 1 LmjF.24.0370 GO:0009058        biosynthetic process Biological Process
## 2 LmjF.24.0370 GO:0005737                   cytoplasm Cellular Component
## 3 LmjF.24.0370 GO:0031981               nuclear lumen Cellular Component
## 4 LmjF.24.0370 GO:0097014               ciliary plasm Cellular Component
## 5 LmjF.24.0370 GO:0030170 pyridoxal phosphate binding Molecular Function
## 6 LmjF.34.2180 GO:0005488                     binding Molecular Function</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" title="1"><span class="co"># retrieve KEGG, etc. pathway annotations</span></a>
<a class="sourceLine" id="cb62-2" title="2">gene_pathway_mapping &lt;-<span class="st"> </span><span class="kw">select</span>(orgdb, <span class="dt">keys=</span>gids, <span class="dt">keytype=</span><span class="st">'GID'</span>,</a>
<a class="sourceLine" id="cb62-3" title="3">                               <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">'PATHWAY'</span>, <span class="st">'PATHWAY_SOURCE'</span>))</a></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" title="1"><span class="kw">table</span>(gene_pathway_mapping<span class="op">$</span>PATHWAY_SOURCE)</a></code></pre></div>
<pre><code>## 
##       KEGG   LeishCyc    MetaCyc TrypanoCyc 
##      15803       2367      66680       2475</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" title="1"><span class="kw">head</span>(gene_pathway_mapping)</a></code></pre></div>
<pre><code>##            GID                                                   PATHWAY
## 1 LmjF.24.0370                  L-asparagine degradation III (mammalian)
## 2 LmjF.24.0370                                 L-aspartate degradation I
## 3 LmjF.24.0370                                    aspartate biosynthesis
## 4 LmjF.24.0370                                  L-aspartate biosynthesis
## 5 LmjF.24.0370                                    aspartate biosynthesis
## 6 LmjF.24.0370 superpathway of L-aspartate and L-asparagine biosynthesis
##   PATHWAY_SOURCE
## 1        MetaCyc
## 2        MetaCyc
## 3       LeishCyc
## 4        MetaCyc
## 5     TrypanoCyc
## 6        MetaCyc</code></pre>
</div>
<div id="working-with-eupathdb-granges-resources" class="section level1">
<h1>Working with EuPathDB GRanges resources</h1>
<p>In addition to retrieving gene annotations, AnnotationHub can also be used to query GenomicRange (GRange) objects containing information about gene and transcript structure.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" title="1"><span class="co"># query AnnotationHub</span></a>
<a class="sourceLine" id="cb68-2" title="2">res &lt;-<span class="st"> </span><span class="kw">query</span>(ah, <span class="kw">c</span>(<span class="st">'Leishmania major strain Friedlin'</span>, <span class="st">'GRanges'</span>, <span class="st">'EuPathDB'</span>))</a>
<a class="sourceLine" id="cb68-3" title="3">res</a></code></pre></div>
<pre><code>## AnnotationHub with 1 record
## # snapshotDate(): 2018-10-24 
## # names(): AH65354
## # $dataprovider: TriTrypDB
## # $species: Leishmania major
## # $rdataclass: GRanges
## # $rdatadateadded: 2018-10-15
## # $title: Leishmania major strain Friedlin transcript information
## # $description: TriTrypDB 39 transcript information for Leishmania majo...
## # $taxonomyid: 347515
## # $genome: TriTrypDB-39_LmajorFriedlin
## # $sourcetype: GFF
## # $sourceurl: http://TriTrypDB.org/common/downloads/Current_Release/Lma...
## # $sourcesize: NA
## # $tags:
## #   c(&quot;Annotation,EuPathDB,Eukaryote,Pathogen,Parasite,Trypanosome,Kinetoplastid,Leishmania&quot;,
## #   &quot;EuPathDB&quot;) 
## # retrieve record with 'object[[&quot;AH65354&quot;]]'</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" title="1"><span class="co"># retrieve a GRanges instance associated with the result record</span></a>
<a class="sourceLine" id="cb70-2" title="2">gr &lt;-<span class="st"> </span>res[[<span class="st">'AH65354'</span>]]</a></code></pre></div>
<pre><code>## downloading 0 resources</code></pre>
<pre><code>## loading from cache 
##     '/home/trey//.AnnotationHub/72100'</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb73-1" title="1"><span class="kw">summary</span>(gr)</a></code></pre></div>
<pre><code>## [1] &quot;GRanges object with 37311 ranges and 9 metadata columns&quot;</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" title="1"><span class="kw">head</span>(gr)</a></code></pre></div>
<pre><code>## GRanges object with 6 ranges and 9 metadata columns:
##       seqnames        ranges strand |   source     type     score
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;
##   [1]  LmjF.24 117748-119043      + | EuPathDB     gene      &lt;NA&gt;
##   [2]  LmjF.24 117748-119043      + | EuPathDB     mRNA      &lt;NA&gt;
##   [3]  LmjF.24 117748-119043      + | EuPathDB     exon      &lt;NA&gt;
##   [4]  LmjF.24 117748-119043      + | EuPathDB      CDS      &lt;NA&gt;
##   [5]  LmjF.34 961963-966600      - | EuPathDB     gene      &lt;NA&gt;
##   [6]  LmjF.34 961963-966600      - | EuPathDB     mRNA      &lt;NA&gt;
##           phase                        ID
##       &lt;integer&gt;               &lt;character&gt;
##   [1]      &lt;NA&gt;              LmjF.24.0370
##   [2]      &lt;NA&gt;         LmjF.24.0370:mRNA
##   [3]      &lt;NA&gt;      exon_LmjF.24.0370-E1
##   [4]         0 LmjF.24.0370:mRNA-p1-CDS1
##   [5]      &lt;NA&gt;              LmjF.34.2180
##   [6]      &lt;NA&gt;         LmjF.34.2180:mRNA
##                                description            Parent
##                                &lt;character&gt;   &lt;CharacterList&gt;
##   [1] aspartate aminotransferase, putative              &lt;NA&gt;
##   [2] aspartate aminotransferase, putative      LmjF.24.0370
##   [3]                                 &lt;NA&gt; LmjF.24.0370:mRNA
##   [4]                                 &lt;NA&gt; LmjF.24.0370:mRNA
##   [5]      hypothetical protein, conserved              &lt;NA&gt;
##   [6]      hypothetical protein, conserved      LmjF.34.2180
##                                   Note    protein_source_id
##                        &lt;CharacterList&gt;          &lt;character&gt;
##   [1]                             &lt;NA&gt;                 &lt;NA&gt;
##   [2] 2.6.1.1 (Aspartate transaminase)                 &lt;NA&gt;
##   [3]                             &lt;NA&gt;                 &lt;NA&gt;
##   [4]                             &lt;NA&gt; LmjF.24.0370:mRNA-p1
##   [5]                             &lt;NA&gt;                 &lt;NA&gt;
##   [6]                             &lt;NA&gt;                 &lt;NA&gt;
##   -------
##   seqinfo: 36 sequences from TriTrypDB-39_LmajorFriedlin genome; no seqlengths</code></pre>
<p>The resulting <code>GRanges</code> object can then be interacted with using the <a href="https://bioconductor.org/packages/3.7/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.pdf">standard GRanges functions</a>, including:</p>
<ul>
<li>seqnames</li>
<li>strand</li>
<li>width</li>
</ul>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb77-1" title="1"><span class="co"># chromosome names</span></a>
<a class="sourceLine" id="cb77-2" title="2"><span class="kw">seqnames</span>(gr)</a></code></pre></div>
<pre><code>## factor-Rle of length 37311 with 9006 runs
##   Lengths:       4       4       4       4 ...       4       4       4
##   Values : LmjF.24 LmjF.34 LmjF.36 LmjF.28 ... LmjF.04 LmjF.11 LmjF.27
## Levels(36): LmjF.01 LmjF.02 LmjF.03 LmjF.04 ... LmjF.34 LmjF.35 LmjF.36</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb79-1" title="1"><span class="co"># strand information</span></a>
<a class="sourceLine" id="cb79-2" title="2"><span class="kw">strand</span>(gr)</a></code></pre></div>
<pre><code>## factor-Rle of length 37311 with 4630 runs
##   Lengths:  4 12  4  4  4  4  3  4 31  4 ...  8  8  4  4  4  4  4 23  4  4
##   Values :  +  -  +  -  +  -  +  -  +  - ...  +  -  +  -  +  -  +  -  +  -
## Levels(3): + - *</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb81-1" title="1"><span class="co"># feature widths</span></a>
<a class="sourceLine" id="cb81-2" title="2"><span class="kw">head</span>(<span class="kw">width</span>(gr))</a></code></pre></div>
<pre><code>## [1] 1296 1296 1296 1296 4638 4638</code></pre>
<p>Some information can be retrieve directly as object properties using the <code>$</code> operator:</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb83-1" title="1"><span class="co"># list of location types in the resource</span></a>
<a class="sourceLine" id="cb83-2" title="2"><span class="kw">table</span>(gr<span class="op">$</span>type)</a></code></pre></div>
<pre><code>## 
##            gene            mRNA            exon             CDS 
##            9378            8519            9583            8606 
##          snoRNA            rRNA three_prime_UTR  five_prime_UTR 
##             741              63               8             241 
##           ncRNA            tRNA           snRNA 
##              83              83               6</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb85-1" title="1"><span class="kw">table</span>(gr<span class="op">@</span>strand)</a></code></pre></div>
<pre><code>## 
##     +     -     * 
## 20366 16945     0</code></pre>
<p>To subset the GRanges instance, you can use the standard <code>[</code> operator:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb87-1" title="1"><span class="co"># get the first three ranges</span></a>
<a class="sourceLine" id="cb87-2" title="2">gr[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]</a></code></pre></div>
<pre><code>## GRanges object with 3 ranges and 9 metadata columns:
##       seqnames        ranges strand |   source     type     score
##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;numeric&gt;
##   [1]  LmjF.24 117748-119043      + | EuPathDB     gene      &lt;NA&gt;
##   [2]  LmjF.24 117748-119043      + | EuPathDB     mRNA      &lt;NA&gt;
##   [3]  LmjF.24 117748-119043      + | EuPathDB     exon      &lt;NA&gt;
##           phase                   ID                          description
##       &lt;integer&gt;          &lt;character&gt;                          &lt;character&gt;
##   [1]      &lt;NA&gt;         LmjF.24.0370 aspartate aminotransferase, putative
##   [2]      &lt;NA&gt;    LmjF.24.0370:mRNA aspartate aminotransferase, putative
##   [3]      &lt;NA&gt; exon_LmjF.24.0370-E1                                 &lt;NA&gt;
##                  Parent                             Note protein_source_id
##         &lt;CharacterList&gt;                  &lt;CharacterList&gt;       &lt;character&gt;
##   [1]              &lt;NA&gt;                             &lt;NA&gt;              &lt;NA&gt;
##   [2]      LmjF.24.0370 2.6.1.1 (Aspartate transaminase)              &lt;NA&gt;
##   [3] LmjF.24.0370:mRNA                             &lt;NA&gt;              &lt;NA&gt;
##   -------
##   seqinfo: 36 sequences from TriTrypDB-39_LmajorFriedlin genome; no seqlengths</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb89-1" title="1"><span class="co"># get all gene entries on chromosome 4</span></a>
<a class="sourceLine" id="cb89-2" title="2">chr4_genes &lt;-<span class="st"> </span>gr[gr<span class="op">$</span>type <span class="op">==</span><span class="st"> 'gene'</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw">seqnames</span>(gr) <span class="op">==</span><span class="st"> 'LmjF.04'</span>]</a>
<a class="sourceLine" id="cb89-3" title="3"><span class="kw">summary</span>(chr4_genes)</a></code></pre></div>
<pre><code>## [1] &quot;GRanges object with 130 ranges and 9 metadata columns&quot;</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb91-1" title="1"><span class="co">## Hey, checkit, there are 130 genes on chromosome 4.</span></a></code></pre></div>
</div>
<div id="installing-a-new-species-in-your-r-session" class="section level1">
<h1>Installing a new species in your R session</h1>
<p>When AnnotationHub and the EupathDB web resources get out of sync, one might want to install and manipulate the newest version of the available Eupath data. The following blocks demonstrate how one might do those tasks.</p>
<div id="getting-a-less-popular-tritrypdb-data-set" class="section level2">
<h2>Getting a less popular TriTrypDB data set</h2>
<p>In some of my work, it is interesting to compare genes from species like Leishmania to Crithidia fasculata. In order to get the newest version of that data, we need to query the TriTrypDB web resource rather than EupathDB.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb92-1" title="1"><span class="co">## Note that some but not all web services have moved to https...</span></a>
<a class="sourceLine" id="cb92-2" title="2"><span class="co">## tri_meta &lt;- download_eupath_metadata(webservice=&quot;tritrypdb&quot;)</span></a>
<a class="sourceLine" id="cb92-3" title="3"></a>
<a class="sourceLine" id="cb92-4" title="4">sc_entry &lt;-<span class="st"> </span><span class="kw">get_eupath_entry</span>(<span class="dt">species=</span><span class="st">&quot;cerevisiae&quot;</span>, <span class="dt">webservice=</span><span class="st">&quot;fungidb&quot;</span>)</a></code></pre></div>
<pre><code>## Warning: 'BiocInstaller' and 'biocLite()' are deprecated, use the 'BiocManager'
##   CRAN package instead.</code></pre>
<pre><code>## Added 21 species after using only the genus species.</code></pre>
<pre><code>## Unable to find species names for 2 species.</code></pre>
<pre><code>## Candida auris, Candida glabrata</code></pre>
<pre><code>## Found: Saccharomyces cerevisiae</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb98-1" title="1">sc_name &lt;-<span class="st"> </span>sc_entry[[<span class="st">&quot;Species&quot;</span>]]</a>
<a class="sourceLine" id="cb98-2" title="2">sc_entry</a></code></pre></div>
<pre><code>##     BiocVersion                      Genome NumGenes NumOrthologs
## 128         3.8 FungiDB-43_ScerevisiaeS288c     6350         5892
##     SourceType
## 128        GFF
##                                                                                                    SourceUrl
## 128 http://FungiDB.org/common/downloads/release-43/ScerevisiaeS288c/gff/data/FungiDB-43_ScerevisiaeS288c.gff
##     SourceVersion                  Species TaxonomyId Coordinate_1_based
## 128            43 Saccharomyces cerevisiae     559292               TRUE
##     DataProvider                       Maintainer
## 128      FungiDB Keith Hughitt &lt;khughitt@umd.edu&gt;
##                                                             Tags
## 128 Annotation:EuPathDB:Eukaryote:Pathogen:Parasite:Fungus:Fungi
##                                     BsgenomePkg
## 128 BSGenome.Saccharomyces.cerevisiae.S288c.v43
##                                         GrangesPkg
## 128 GRanges.Saccharomyces.cerevisiae.S288c.v43.rda
##                                 OrganismdbiPkg
## 128 fungidb.Saccharomyces.cerevisiae.S288c.v43
##                            OrgdbPkg
## 128 org.Scerevisiae.S288c.v43.eg.db
##                                             TxdbPkg Strain
## 128 TxDb.Saccharomyces.cerevisiae.S288c.FungiDB.v43  S288c
##                              Taxon Genus Sp                TaxonUnmodified
## 128 Saccharomyces cerevisiae S288c          Saccharomyces cerevisiae S288c</code></pre>
<div id="creating-the-package" class="section level3">
<h3>Creating the package</h3>
<p>Now that we have the canonical name from the TriTrypDB for Crithidia, we can create a fresh orgdb package for it. I will not actually run this because it takes a long time and prints a lot to screen.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb100-1" title="1">orgdb_pkg &lt;-<span class="st"> </span><span class="kw">make_eupath_orgdb</span>(sc_entry, <span class="dt">reinstall=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb100-2" title="2">txdb_pkg &lt;-<span class="st"> </span><span class="kw">make_eupath_txdb</span>(sc_entry)</a>
<a class="sourceLine" id="cb100-3" title="3">bsgenome_pkg &lt;-<span class="st"> </span><span class="kw">make_eupath_bsgenome</span>(sc_entry)</a>
<a class="sourceLine" id="cb100-4" title="4">organ_pkg &lt;-<span class="st"> </span><span class="kw">make_eupath_organismdbi</span>(sc_entry)</a></code></pre></div>
</div>
<div id="another-example-from-the-tritrypdb" class="section level3">
<h3>Another example from the tritrypdb</h3>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb101-1" title="1">lm_entry &lt;-<span class="st"> </span><span class="kw">get_eupath_entry</span>(<span class="dt">species=</span><span class="st">&quot;Friedlin&quot;</span>, <span class="dt">webservice=</span><span class="st">&quot;tritrypdb&quot;</span>)</a>
<a class="sourceLine" id="cb101-2" title="2">lm_orgdb &lt;-<span class="st"> </span><span class="kw">make_eupath_orgdb</span>(lm_entry, <span class="dt">reinstall=</span><span class="ot">TRUE</span>)</a></code></pre></div>
</div>
</div>
<div id="extracting-annotation-data" class="section level2">
<h2>Extracting annotation data</h2>
<p>The following demonstrates ways to extract data from the generated orgdb package.</p>
<div id="choosing-and-loading-a-generated-package" class="section level3">
<h3>Choosing and loading a generated package</h3>
<p>Because the eupathdb is constantly evolving, the get_eupath_pkgnames() function will use the metadata generated in download_eupath_metadata() in order to provide a differently names package for each eupathdb revision.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb102-1" title="1">orgdb_pkg &lt;-<span class="st"> </span><span class="kw">get_eupath_pkgnames</span>(sc_entry)</a>
<a class="sourceLine" id="cb102-2" title="2">sc_orgdb &lt;-<span class="st"> </span>orgdb_pkg<span class="op">$</span>orgdb</a>
<a class="sourceLine" id="cb102-3" title="3"><span class="co">## Here is the name of the current yeast package.</span></a>
<a class="sourceLine" id="cb102-4" title="4">sc_orgdb</a></code></pre></div>
<pre><code>## org.Scerevisiae.v43.eg.db</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb104-1" title="1"><span class="co">## Thus we see the v41 (as of late 2018), a number which presumably will continue increasing.</span></a>
<a class="sourceLine" id="cb104-2" title="2"><span class="co">## We can set the version parameter to change this if we have a previous version installed.</span></a>
<a class="sourceLine" id="cb104-3" title="3"></a>
<a class="sourceLine" id="cb104-4" title="4"><span class="co">## Now get the set of available columns from it:</span></a>
<a class="sourceLine" id="cb104-5" title="5"><span class="kw">library</span>(sc_orgdb, <span class="dt">character=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>## </code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb106-1" title="1">avail_columns &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">columns</span>(<span class="kw">get0</span>(sc_orgdb))</a>
<a class="sourceLine" id="cb106-2" title="2"><span class="kw">head</span>(avail_columns)</a></code></pre></div>
<pre><code>## [1] &quot;ANNOT_ANNOTATED_GO_COMPONENT&quot;    &quot;ANNOT_ANNOTATED_GO_FUNCTION&quot;    
## [3] &quot;ANNOT_ANNOTATED_GO_ID_COMPONENT&quot; &quot;ANNOT_ANNOTATED_GO_ID_FUNCTION&quot; 
## [5] &quot;ANNOT_ANNOTATED_GO_ID_PROCESS&quot;   &quot;ANNOT_ANNOTATED_GO_PROCESS&quot;</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb108-1" title="1"><span class="co">## There are lots of columns!</span></a>
<a class="sourceLine" id="cb108-2" title="2"><span class="kw">length</span>(avail_columns)</a></code></pre></div>
<pre><code>## [1] 111</code></pre>
</div>
<div id="extracting-data-from-a-package" class="section level3">
<h3>Extracting data from a package</h3>
<p>Once we have a package loaded, everything else is an application of the AnnotationDbi interface. Here are a few examples.</p>
<p>Later I will show a couple of shortcuts for these methods.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb110-1" title="1"><span class="co">## The columns which begin with strings like 'PATHWAY' or 'INTERPRO' are actually separate</span></a>
<a class="sourceLine" id="cb110-2" title="2"><span class="co">## sql tables in the orgdb database, and as such will lead to a hugely redundant data table</span></a>
<a class="sourceLine" id="cb110-3" title="3"><span class="co">## if we select them.</span></a>
<a class="sourceLine" id="cb110-4" title="4">chosen_columns_idx &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="dt">x=</span>avail_columns, <span class="dt">pattern=</span><span class="st">&quot;^ANNOT&quot;</span>)</a>
<a class="sourceLine" id="cb110-5" title="5">chosen_columns &lt;-<span class="st"> </span>avail_columns[chosen_columns_idx]</a>
<a class="sourceLine" id="cb110-6" title="6"></a>
<a class="sourceLine" id="cb110-7" title="7"><span class="co">## Now we have a set of columns of interest, let us get a data table/data frame.</span></a>
<a class="sourceLine" id="cb110-8" title="8">sc_annot &lt;-<span class="st"> </span><span class="kw">load_orgdb_annotations</span>(<span class="dt">orgdb=</span>sc_orgdb, <span class="dt">keytype=</span><span class="st">&quot;gid&quot;</span>, <span class="dt">fields=</span>chosen_columns)</a></code></pre></div>
<pre><code>## Unable to find CDSNAME, setting it to ANNOT_GENE_NAME.</code></pre>
<pre><code>## Unable to find CDSCHROM in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSSTRAND in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSSTART in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSEND in the db, removing it.</code></pre>
<pre><code>## Extracted all gene ids.</code></pre>
<pre><code>## Attempting to select: ANNOT_GENE_NAME, GENE_TYPE, ANNOT_ANNOTATED_GO_COMPONENT, ANNOT_ANNOTATED_GO_FUNCTION, ANNOT_ANNOTATED_GO_ID_COMPONENT, ANNOT_ANNOTATED_GO_ID_FUNCTION, ANNOT_ANNOTATED_GO_ID_PROCESS, ANNOT_ANNOTATED_GO_PROCESS, ANNOT_CDS, ANNOT_CDS_LENGTH, ANNOT_CHROMOSOME, ANNOT_EC_NUMBERS, ANNOT_EC_NUMBERS_DERIVED, ANNOT_EXON_COUNT, ANNOT_FIVE_PRIME_UTR_LENGTH, ANNOT_GENE_ENTREZ_ID, ANNOT_GENE_EXON_COUNT, ANNOT_GENE_HTS_NONCODING_SNPS, ANNOT_GENE_HTS_NONSYN_SYN_RATIO, ANNOT_GENE_HTS_NONSYNONYMOUS_SNPS, ANNOT_GENE_HTS_STOP_CODON_SNPS, ANNOT_GENE_HTS_SYNONYMOUS_SNPS, ANNOT_GENE_LOCATION_TEXT, ANNOT_GENE_NAME, ANNOT_GENE_ORTHOLOG_NUMBER, ANNOT_GENE_ORTHOMCL_NAME, ANNOT_GENE_PARALOG_NUMBER, ANNOT_GENE_PREVIOUS_IDS, ANNOT_GENE_PRODUCT, ANNOT_GENE_SOURCE_ID, ANNOT_GENE_TOTAL_HTS_SNPS, ANNOT_GENE_TRANSCRIPT_COUNT, ANNOT_GENE_TYPE, ANNOT_HAS_MISSING_TRANSCRIPTS, ANNOT_INTERPRO_DESCRIPTION, ANNOT_INTERPRO_ID, ANNOT_IS_PSEUDO, ANNOT_ISOELECTRIC_POINT, ANNOT_LOCATION_TEXT, ANNOT_MATCHED_RESULT, ANNOT_MOLECULAR_WEIGHT, ANNOT_ORGANISM, ANNOT_PFAM_DESCRIPTION, ANNOT_PFAM_ID, ANNOT_PIRSF_DESCRIPTION, ANNOT_PIRSF_ID, ANNOT_PREDICTED_GO_COMPONENT, ANNOT_PREDICTED_GO_FUNCTION, ANNOT_PREDICTED_GO_ID_COMPONENT, ANNOT_PREDICTED_GO_ID_FUNCTION, ANNOT_PREDICTED_GO_ID_PROCESS, ANNOT_PREDICTED_GO_PROCESS, ANNOT_PROJECT_ID, ANNOT_PROSITEPROFILES_DESCRIPTION, ANNOT_PROSITEPROFILES_ID, ANNOT_PROTEIN_LENGTH, ANNOT_PROTEIN_SEQUENCE, ANNOT_SEQUENCE_ID, ANNOT_SIGNALP_PEPTIDE, ANNOT_SIGNALP_SCORES, ANNOT_SMART_DESCRIPTION, ANNOT_SMART_ID, ANNOT_SOURCE_ID, ANNOT_STRAND, ANNOT_SUPERFAMILY_DESCRIPTION, ANNOT_SUPERFAMILY_ID, ANNOT_THREE_PRIME_UTR_LENGTH, ANNOT_TIGRFAM_DESCRIPTION, ANNOT_TIGRFAM_ID, ANNOT_TM_COUNT, ANNOT_TRANS_FOUND_PER_GENE_INTERNAL, ANNOT_TRANSCRIPT_INDEX_PER_GENE, ANNOT_TRANSCRIPT_LENGTH, ANNOT_TRANSCRIPT_LINK, ANNOT_TRANSCRIPT_PRODUCT, ANNOT_TRANSCRIPT_SEQUENCE, ANNOT_TRANSCRIPTS_FOUND_PER_GENE, ANNOT_UNIPROT_ID, ANNOT_WDK_WEIGHT</code></pre>
<pre><code>## 'select()' returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb119-1" title="1"><span class="co">## load_orgdb_annotations will fill out separate dataframes for each annotation type,</span></a>
<a class="sourceLine" id="cb119-2" title="2"><span class="co">## genes, exons, transcripts, etc.  In this case, we only want the genes</span></a>
<a class="sourceLine" id="cb119-3" title="3"><span class="co">## (The eupathdb does not provide much information for the others.)</span></a>
<a class="sourceLine" id="cb119-4" title="4">sc_genes &lt;-<span class="st"> </span>sc_annot[[<span class="st">&quot;genes&quot;</span>]]</a>
<a class="sourceLine" id="cb119-5" title="5"><span class="kw">dim</span>(sc_genes)</a></code></pre></div>
<pre><code>## [1] 6445   79</code></pre>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb121-1" title="1"><span class="kw">head</span>(sc_genes)</a></code></pre></div>
<pre><code>##         gid annot_gene_name      gene_type
## Q0045 Q0045            COX1 protein coding
## Q0050 Q0050             AI1 protein coding
## Q0055 Q0055             AI2 protein coding
## Q0060 Q0060             AI3 protein coding
## Q0065 Q0065             AI4 protein coding
## Q0070 Q0070       AI5_ALPHA protein coding
##                                                                  annot_annotated_go_component
## Q0045 integral component of membrane;mitochondrial respiratory chain complex IV;mitochondrion
## Q0050                                                                           mitochondrion
## Q0055                                                                           mitochondrion
## Q0060                                mitochondrial respiratory chain complex IV;mitochondrion
## Q0065                                mitochondrial respiratory chain complex IV;mitochondrion
## Q0070                                mitochondrial respiratory chain complex IV;mitochondrion
##                               annot_annotated_go_function
## Q0045                       cytochrome-c oxidase activity
## Q0050                RNA-directed DNA polymerase activity
## Q0055                RNA-directed DNA polymerase activity
## Q0060 cytochrome-c oxidase activity;endonuclease activity
## Q0065 cytochrome-c oxidase activity;endonuclease activity
## Q0070 cytochrome-c oxidase activity;endonuclease activity
##        annot_annotated_go_id_component annot_annotated_go_id_function
## Q0045 GO:0016021;GO:0005751;GO:0005739                     GO:0004129
## Q0050                       GO:0005739                     GO:0003964
## Q0055                       GO:0005739                     GO:0003964
## Q0060            GO:0005751;GO:0005739          GO:0004129;GO:0004519
## Q0065            GO:0005751;GO:0005739          GO:0004129;GO:0004519
## Q0070            GO:0005751;GO:0005739          GO:0004129;GO:0004519
##                                annot_annotated_go_id_process
## Q0045                       GO:0009060;GO:0015990;GO:0006123
## Q0050            GO:0006278;GO:0006315;GO:0006397;GO:0090615
## Q0055                       GO:0006315;GO:0006314;GO:0090615
## Q0060            GO:0009060;GO:0008150;GO:0015990;GO:0006123
## Q0065 GO:0008380;GO:0009060;GO:0015990;GO:0006314;GO:0006123
## Q0070            GO:0009060;GO:0015990;GO:0006123;GO:0006316
##                                                                                                                                annot_annotated_go_process
## Q0045                            aerobic respiration;electron transport coupled proton transport;mitochondrial electron transport, cytochrome c to oxygen
## Q0050                                     RNA-dependent DNA biosynthetic process;homing of group II introns;mRNA processing;mitochondrial mRNA processing
## Q0055                                                                              homing of group II introns;intron homing;mitochondrial mRNA processing
## Q0060         aerobic respiration;biological_process;electron transport coupled proton transport;mitochondrial electron transport, cytochrome c to oxygen
## Q0065 RNA splicing;aerobic respiration;electron transport coupled proton transport;intron homing;mitochondrial electron transport, cytochrome c to oxygen
## Q0070 aerobic respiration;electron transport coupled proton transport;mitochondrial electron transport, cytochrome c to oxygen;movement of group I intron
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   annot_cds
## Q0045                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCACCCTGAAGTATATATTTTAATTATTCCTGGATTTGGTATTATTTCACATGTAGTATCAACATATTCTAAAAAACCTGTATTTGGTGAAATTTCAATGGTATATGCTATGGCTTCAATTGGATTATTAGGATTCTTAGTATGATCACATCATATGTATATTGTAGGATTAGATGCAGATCTTAGAGCATATTTCCTATCTGCACTAATGATTATTGCAATTCCAACAGGAATTAAAATTTTCTCATGATTAGCTCTAATCCATGGTGGTTCAATTAGATTAGCACTACCTATGTTATATGCAATTGCATTCTTATTCTTATTCACAATGGGTGGTTTAACTGGTGTTGCCTTAGCTAACGCCTCATTAGATGTAGCATTCCACGATACTTACTACGTGGTGGGACATTTTCACTATGTATTATCAATGGGTGCTATTTTCTCTTTATTTGCAGGATACTATTATTGAAGTCCTCAAATTTTAGGTTTAAACTATAATGAAAAATTAGCTCAAATTCAATTCTGATTAATTTTCATTGGGGCTAATGTTATTTTCTTCCCAATGCATTTTTTAGGTATTAATGGTATGCCTAGAAGAATTCCTGATTATCCTGATGCTTTCGCAGGATGAAATTATGTCGCTTCTATTGGTTCATTCATTGCACTATTATCATTATTCTTATTTATCTATATTTTATATGATCAATTAGTTAATGGATTAAACAATAAAGTTAATAATAAATCAGTTATTTATAATAAAGCACCTGATTTTGTAGAATCTAATCTTATCTTTAATTTAAATACAGTTAAATCTTCATCTATCGAATTCTTATTAACTTCTCCACCAGCTGTACACTCATTTAATACACCAGCTGTACAATCTTAA
## Q0050                                                             ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGGTGCGCCTCTCAGTGCGTATATTTCGTTGATGCGTCTAGCATTAGTATTATGAATCATCAATAGATACTTAAAACATATGACTAACTCAGTAGGGGCTAACTTTACGGGGACAATAGCATGTCATAAAACACCTATGATTAGTGTAGGTGGAGTTAAGTGTTACATGGTTAGGTTAACGAACTTCTTACAAGTCTTTATCAGGATTACAATTTCCTCTTATCATTTGGATATAGTAAAACAAGTTTGATTATTTTACGTTGAGGTAATCAGATTATGATTCATTGTTTTAGATAGCACAGGCAGTGTGAAAAAGATGAAGGACCTAAATAACACAAAAGGAAATACGAAAAGTGAGGGATCAACTGAAAGAGGAAACTCTGGAGTTGACAGAGGTATAGTAGTACCGAATACTCAAATAAAAATGAGATTTTTAAATCAAGTTAGATACTATTCAGTAAATAATAATTTAAAAATAGGGAAGGATACCAATATTGAGTTATCAAAAGATACAAGTACTTCGGACTTGTTAGAATTTGAGAAATTAGTAATAGATAATATAAATGAGGAAAATATAAATAATAATTTATTAAGTATTATAAAAAACGTAGATATATTAATATTAGCATATAATAGAATTAAGAGTAAACCTGGTAATATAACTCCAGGTACAACATTAGAAACATTAGATGGTATAAATATAATATATTTAAATAAATTATCAAATGAATTAGGAACAGGTAAATTCAAATTTAAACCCATGAGAATAGTTAATATTCCTAAACCTAAAGGTGGTATAAGACCTTTAAGTGTAGGTAATCCAAGAGATAAAATTGTACAAGAAGTTATAAGAATAATTTTAGATACAATTTTTGATAAAAAGATATCAACACATTCACATGGTTTTAGAAAGAATATAAGTTGTCAAACAGCAATTTGAGAAGTTAGAAATATATTTGGTGGAAGTAATTGATTTATTGAAGTAGACTTAAAAAAATGTTTTGATACAATTTCTCATGATTTAATTATTAAAGAATTAAAAAGATATATTTCAGATAAAGGTTTTATTGATTTAGTATATAAATTATTAAGAGCTGGTTATATTGATGAGAAAGGAACTTATCATAAACCTATATTAGGTTTACCTCAAGGATCATTAATTAGTCCTATCTTATGTAATATTGTAATAACATTGGTAGATAATTGATTAGAAGATTATATTAATTTATATAATAAAGGTAAAGTTAAAAAACAACATCCTACATATAAAAAATTATCAAGAATAATTGCAAAAGCTAAAATATTTTCGACAAGATTAAAATTACATAAAGAAAGAGCTAAAGGCCCACTATTTATTTATAATGATCCTAATTTCAAGAGAATAAAATACGTTAGATATGCAGATGATATTTTAATTGGGGTATTAGGTTCAAAAAATGATTGTAAAATAATCAAAAGAGATTTAAACAATTTTTTAAATTCATTAGGTTTAACTATAAATGAAGAAAAAACTTTAATTACTTGTGCAACTGAACTACCAGCAAGATTTTTAGGTTATAATATTTCAATTACACCTTTAAAAAGAATACCTACAGTTACTAAACTAATTAGAGGTAAACTTATTAGAAGTAGAAATACAACTAGACCTATTATTAATGCACCAATTAGAGATATTATCAATAAATTAGCTACTAATGGATATTGTAAGCATAATAAAAATGGTAGAATAGGAGTGCCTACAAGAGTAGGTAGATGACTATATGAAGAACCTAGAACAATTATTAATAATTATAAAGCGTTAGGTAGAGGTATCTTAAATTATTATAAATTAGCTACTAATTATAAAAGATTAAGAGAAAGAATCTATTACGTATTATATTATTCATGTGTATTAACTTTAGCTAGTAAATATAGATTAAAAACAATAAGTAAAACTATTAAAAAATTTGGTTATAATTTAAATATTATTGAAAATGATAAATTAATTGCCAATTTTCCAAGAAATACTTTTGATAATATCAAAAAAATTGAAAATCATGGTATATTTATATATATATCAGAAGCTAAAGTAACTGATCCTTTTGAATATATCGATTCAATTAAATATATATTACCTACAGCTAAAGCTAATTTTAATAAACCTTGTAGTATTTGTAATTCAACTATTGATGTAGAAATACATCATGTTAAACAATTACATAGAGGTATATTAAAAGCACTTAAAGATTATATTCTAGGTAGAATAATTACCATAAACAGAAAACAAATTCCATTATGTAAACAATGTCATATTAAAACACATAAAAATAAATTTAAAAATATAGGACCTGGTATATAA
## Q0055 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTGTGCGCCGTTTCGCTTAATTTATCACTGTATTGAAGTGTTAATTGATAAACATATCTCTGTTTATTCAATTAATGAAAACTTTACCGTATCATTTTGGTTCTGATTATTAGTAGTAACATACATAGTATTTAGATACGTAAACCATATGGCTTACCCAGTTGGGGCCAACTCAACGGGGACAATAGCATGCCATAAAAGCGCTGGAGTAAAACAGCCAGCGCAAGGTAAGAACTGTCCGATGGCTAGGTTAACGAATTCCTGTAAAGAATGTTTAGGGTTCTCATTAACTCCTTCCCACTTGGGGATTGTGATTCATGCTTATGTATTGGAAGAAGAGGTACACGAGTTAACCAAAAATGAATCATTAGCTTTAAGTAAAAGTTGACATTTGGAGGGCTGTACGAGTTCAAATGGAAAATTAAGAAATACGGGATTGTCCGAAAGGGGAAACCCTGGGGATAACGGAGTCTTCATAGTACCCAAATTTAATTTAAATAAAGTGAGATACTTTAGTACTTTATCTAAATTAAATGCAAGGAAGGAAGACAGTTTAGCGTATTTAACAAAGATTAATACTACGGATTTTTCCGAGTTAAATAAATTAATAGAAAATAATCATAATAAACTTGAAACCATTAATACTAGAATTTTAAAATTAATGTCAGATATTAGAATGTTATTAATTGCTTATAATAAAATTAAAAGTAAGAAAGGTAATATATCTAAAGGTTCTAATAATATTACCTTAGATGGGATTAATATTTCATATTTAAATAAATTATCTAAAGATATTAACACTAATATGTTTAAATTTTCTCCGGTTAGAAGAGTTGAAATTCCTAAAACATCTGGAGGATTTAGACCTTTAAGTGTTGGAAATCCTAGAGAAAAAATTGTACAAGAAAGTATGAGAATAATATTAGAAATTATCTATAATAATAGTTTCTCTTATTATTCTCATGGATTTAGACCTAACTTATCTTGTTTAACAGCTATTATTCAATGTAAAAATTATATGCAATACTGTAATTGATTTATTAAAGTAGATTTAAATAAATGCTTTGATACAATTCCACATAATATGTTAATTAATGTATTAAATGAGAGAATCAAAGATAAAGGTTTCATAGACTTATTATATAAATTATTAAGAGCTGGATATGTTGATAAAAATAATAATTATCATAATACAACTTTAGGAATTCCTCAAGGTAGTGTTGTCAGTCCTATTTTATGTAATATTTTTTTAGATAAATTAGATAAATATTTAGAAAATAAATTTGAGAATGAATTCAATACTGGAAATATGTCTAATAGAGGTAGAAATCCAATTTATAATAGTTTATCATCTAAAATTTATAGATGTAAATTATTATCTGAAAAATTAAAATTGATTAGATTAAGAGACCATTACCAAAGAAATATGGGATCTGATAAAAGTTTTAAAAGAGCTTATTTTGTTAGATATGCTGATGATATTATCATTGGTGTAATGGGTTCTCATAATGATTGTAAAAATATTTTAAACGATATTAATAACTTCTTAAAAGAAAATTTAGGTATGTCAATTAATATAGATAAATCCGTTATTAAACATTCTAAAGAAGGAGTTAGTTTTTTAGGGTATGATGTAAAAGTTACACCTTGAGAAAAAAGACCTTATAGAATGATTAAAAAAGGTGATAATTTTATTAGGGTTAGACATCATACTAGTTTAGTTGTTAATGCCCCTATTAGAAGTATTGTAATAAAATTAAATAAACATGGCTATTGTTCTCATGGTATTTTAGGAAAACCCAGAGGGGTTGGAAGATTAATTCATGAAGAAATGAAAACCATTTTAATGCATTACTTAGCTGTTGGTAGAGGTATTATAAACTATTATAGATTAGCTACCAATTTTACCACATTAAGAGGTAGAATTACATACATTTTATTTTATTCATGTTGTTTAACATTAGCAAGAAAATTTAAATTAAATACTGTTAAGAAAGTTATTTTAAAATTCGGTAAAGTATTAGTTGATCCTCATTCAAAAGTTAGTTTTAGTATTGATGATTTTAAAATTAGACATAAAATAAATATAACTGATTCTAATTATACACCTGATGAAATTTTAGATAGATATAAATATATGTTACCTAGATCTTTATCATTATTTAGTGGTATTTGTCAAATTTGTGGTTCTAAACATGATTTAGAAGTACATCACGTAAGAACATTAAATAATGCTGCCAATAAAATTAAAGATGATTATTTATTAGGTAGAATGATTAAGATAAATAGAAAACAAATTACTATCTGTAAAACATGTCATTTTAAAGTTCATCAAGGTAAATATAATGGTCCAGGTTTATAA
## Q0060                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACCAAAAAAGATATGAAAGTAATAATAATAATAATCAAGTAATAGAAAATAAAGAATATAATTTAAAATTAAATTATGATAAGTTGGGACCTTATTTAGCTGGATTAATTGAAGGTGATGGAACTATTCTAGTTCAAAATTCATCTTCAATAAAAAAATCTAAATATAGACCGTTAATTGTTGTAGTATTTAAATTAGAAGATTTAGAATTAGCTAATTATTTATGTAATTTAACTAAATGTGGAAAAGTGTATAAAAAAATTAATCGTAATTATGTATTATGACTTATTCATGATTTAAAAGGTGTATATACATTATTAAATATTATTAATGGATATATGAGAACACCTAAATATGAAGCATTTGTTAGAGGTGCTGAATTTATAAATAATTATATTAATTCAACAACAATTCTACATAATAAATTAAAAAATATAGATAATATTAAAATTAAACCATTAGATACATCAGATATTGGTTCAAACGCTTGATTAGCTGGTATGACAGATGCAGATGGTAATTTTTCTATTAATTTAATAAATGGTAAAAATCGTTCTAGTAGAGCAATGCCTTATTATTGTTTAGAATTAAGACAAAATTATCAAAAAAATTCTAATAATAATAATATTAATTTTTCTTATTTTTATATTATGTCTGCAATTGCACTATATTTTAATGTTAATTTATATAGTAGAGAACGTAATTTAAATTTATTAGTATCTCTTAATAATACGTATAAACTATATTATAGTTATAAAGTAATAGTGGCTAATCTATATAAAAATATTAAAGTAATAGAATACTTTAATAAATATTCTTTATTATCATCTAAACACTTAGATTTTTTAGATTGATCTAAATTAGTTATTTTAATTAATAATGAGGGTCAAAGTATAAAACTTAATGGTAGTTGAGAATTAGGTATAAATTTACGTAAAGATTATAATAAAACTAGAACTACGTTTACTTGATCTCATTTAAAAAATACATATTTAGAAAATAAATAA
## Q0065                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCAAACAGTGGCCCTTATTATTATATTAATAATATATAATGATATGCATTTTTCTAAATGCTGGAAATTATTAAAAAAATGAATTACAAATATTATAAGTCTATTATTTAAAGCCTTATTTGTAAAAATATTCATATCTTATAATAATCAGCAGGATAAGATAATAAATAATCTTATATTAAAAAAAGATAATATTAAAAGATCCTCAGAGACTACAAGAAAAATATTAAATAATTCAATAAATAAAAAATTTAATCAATGATTAGCTGGATTAATTGATGGTGATGGATATTTTGGTATTGTAAGTAAGAAATATGTATCATTAGAAATTCTAGTAGCATTAGAAGATGAAATAGCTTTAAAAGAAATTCAAAATAAATTTGGTGGTTCTATTAAATTAAGATCAGGTGTAAAAGCTATTAGATATAGATTACTTAATAAAACTGGTATAATTAAATTAATTAATGCAGTTAATGGTAATATTAGAAATACTAAAAGATTAGTACAATTTAATAAAGTTTGTATTTTATTAGGTATTGATTTTATTTATCCAATTAAATTAACTAAAGATAATAGTTGATTTGTTGGATTTTTTGATGCTGATGGTACAATTAATTATTCATTTAAAAATAATCATCCTCAATTAACAATTTCTGTAACTAATAAATATTTACAAGATGTACAAGAATATAAAAATATTTTAGGTGGTAATATTTATTTTGATAAATCACAAAATGGTTATTATAAATGATCCATTCAATCAAAAGATATAGTATTAAATTTTATTAATGATTATATTAAAATAAATCCATCAAGAACACTAAAAATAAATAAATTATATTTAAGTAAAGAATTTTATAATTTAAAAGAATTAAAAGCTTATAATAAATCTTCTGATTCAATACAATATAAAGCATGATTAAATTTTGAAAATAAATGAAAAAATAAATAA
## Q0070                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCACCCTGAAGTATATATTTTAATTATTCCTGGATTTGGTATTATTTCACATGTAGTATCAACATATTCTAAAAAACCTGTATTTGGTGAAATTTCAATGGTATATGCTATGGCTTCAATTGGATTATTAGGATTCTTAGTATGATCACATCATATGTATATTGTAGGATTAGATGCAGATCTTAGAGCATATTTCCTATCTGCACTAATGATTATTGCAATTCCAACAGGAATTAAAATTTTCTCATGATTAATAAATCCCTTTAGCAAGGATAAAAATAAAAATAAAAATAAAAAGTTGATCAGAAATTATCAAAAAATAAATAATAATAATATAATAAAAACATATTTAAATAATAATAATATAATTATAATAAATATATATAAAGGTAATTTATATGATATTTATCCAAGATCAAATAGAAATTATATTCAACCAAATAATATTAATAAAGAATTAGTAGTATATGGTTATAATTTAGAATCTTGTGTTGGTATACCTCTATATACTAATATTGTAAAACATATAGTAGGTATTCCTAATAATATTTTATATATTATAACAGGTATTTTATTAACAGATGGTTGAATTGATTATCTATCTAAAAAAGATTTAGATAAAAAAACAATTATAGAAATTAATTGTAGATTTAGATTAAAACAATCAATAATTCATAGTGAATATTTAATATATGTATTTATATTATTATCACATTATTGTATAAGTTATCCTAAAATAAAAATTGCTAAAGTTAAAGGTAAATCATATAATCAATTAGAATTTTATACTAGATCATTACCATGTTTTACTATTTTAAGATATATATTTTATAATGGTAGAGTAAAAATTGTACCTAATAATTTATATGATTTATTAAATTATGAATCTTTAGCTCATATAATTATATGTGATGGTTCATTTGTAAAAGGTGGAGGTTTATATTTAAATTTACAATCTTTTCTAACTAAAGAATTAATTTTTATTATAAATATTTTAAAAATTAAATTTAATTTAAATTGTCTATTACATAAATCTAGAAATAAATATCTTATTTATATAAGAGTAGAATCTGTTAAAAGATTATTTCCTATAATTTATAAATATATTTTACCTTCTATAAGATATAAATTTGATATTATATTATGACAAAAAAAATATAATATGATTAATTAA
##       annot_cds_length annot_chromosome                 annot_ec_numbers
## Q0045           1605.0     Not Assigned   1.9.3.1 (Cytochrome-c oxidase)
## Q0050           2505.0     Not Assigned                                 
## Q0055           2565.0     Not Assigned                                 
## Q0060           1248.0     Not Assigned 3.1.-.- (Acting on ester bonds.)
## Q0065           1671.0     Not Assigned 3.1.-.- (Acting on ester bonds.)
## Q0070           1893.0     Not Assigned 3.1.-.- (Acting on ester bonds.)
##               annot_ec_numbers_derived annot_exon_count
## Q0045   1.9.3.1 (Cytochrome-c oxidase)                8
## Q0050                                                 1
## Q0055                                                 2
## Q0060 3.1.-.- (Acting on ester bonds.)                3
## Q0065 3.1.-.- (Acting on ester bonds.)                4
## Q0070   1.9.3.1 (Cytochrome-c oxidase)                5
##       annot_five_prime_utr_length annot_gene_entrez_id
## Q0045                                           854598
## Q0050                                           854593
## Q0055                                           854594
## Q0060                                           854595
## Q0065                                           854596
## Q0070                                           854597
##       annot_gene_exon_count annot_gene_hts_noncoding_snps
## Q0045                   8.0                           0.0
## Q0050                   1.0                           0.0
## Q0055                   2.0                           0.0
## Q0060                   3.0                           0.0
## Q0065                   4.0                           0.0
## Q0070                   5.0                           0.0
##       annot_gene_hts_nonsyn_syn_ratio annot_gene_hts_nonsynonymous_snps
## Q0045                             0.0                               0.0
## Q0050                             0.0                               0.0
## Q0055                             0.0                               0.0
## Q0060                             0.0                               0.0
## Q0065                             0.0                               0.0
## Q0070                             0.0                               0.0
##       annot_gene_hts_stop_codon_snps annot_gene_hts_synonymous_snps
## Q0045                            0.0                            0.0
## Q0050                            0.0                            0.0
## Q0055                            0.0                            0.0
## Q0060                            0.0                            0.0
## Q0065                            0.0                            0.0
## Q0070                            0.0                            0.0
##         annot_gene_location_text annot_gene_ortholog_number
## Q0045 KP263414:13,818..26,701(+)                       52.0
## Q0050 KP263414:13,818..16,322(+)                        4.0
## Q0055 KP263414:13,818..18,830(+)                        4.0
## Q0060 KP263414:13,818..19,996(+)                       15.0
## Q0065 KP263414:13,818..21,935(+)                        9.0
## Q0070 KP263414:13,818..23,167(+)                       52.0
##       annot_gene_orthomcl_name annot_gene_paralog_number
## Q0045               OG5_128358                       1.0
## Q0050               OG5_137048                       1.0
## Q0055               OG5_137048                       1.0
## Q0060               OG5_194772                       0.0
## Q0065               OG5_156903                       0.0
## Q0070               OG5_128358                       1.0
##       annot_gene_previous_ids                        annot_gene_product
## Q0045                                    cytochrome c oxidase subunit 1
## Q0050                          intron-encoded reverse transcriptase aI1
## Q0055                          intron-encoded reverse transcriptase aI2
## Q0060                               intron-encoded DNA endonuclease aI3
## Q0065                               intron-encoded DNA endonuclease aI4
## Q0070                         intron-encoded DNA endonuclease aI5 alpha
##       annot_gene_source_id annot_gene_total_hts_snps
## Q0045                Q0045                       0.0
## Q0050                Q0050                       0.0
## Q0055                Q0055                       0.0
## Q0060                Q0060                       0.0
## Q0065                Q0065                       0.0
## Q0070                Q0070                       0.0
##       annot_gene_transcript_count annot_gene_type
## Q0045                         1.0  protein coding
## Q0050                         1.0  protein coding
## Q0055                         1.0  protein coding
## Q0060                         1.0  protein coding
## Q0065                         1.0  protein coding
## Q0070                         1.0  protein coding
##       annot_has_missing_transcripts annot_interpro_description
## Q0045                            no                        N/A
## Q0050                            no                        N/A
## Q0055                            no                        N/A
## Q0060                            no                        N/A
## Q0065                            no                        N/A
## Q0070                            no                        N/A
##       annot_interpro_id annot_is_pseudo annot_isoelectric_point
## Q0045               N/A              No                    7.12
## Q0050               N/A              No                   10.44
## Q0055               N/A              No                   10.12
## Q0060               N/A              No                    9.88
## Q0065               N/A              No                   10.14
## Q0070               N/A              No                    9.86
##            annot_location_text annot_matched_result annot_molecular_weight
## Q0045 KP263414:13818..26701(+)                    Y                58797.0
## Q0050 KP263414:13818..16322(+)                    Y                96078.0
## Q0055 KP263414:13818..18830(+)                    Y                98007.0
## Q0060 KP263414:13818..19996(+)                    Y                47916.0
## Q0065 KP263414:13818..21935(+)                    Y                63311.0
## Q0070 KP263414:13818..23167(+)                    Y                72164.0
##            annot_organism
## Q0045 S. cerevisiae S288c
## Q0050 S. cerevisiae S288c
## Q0055 S. cerevisiae S288c
## Q0060 S. cerevisiae S288c
## Q0065 S. cerevisiae S288c
## Q0070 S. cerevisiae S288c
##                                                                annot_pfam_description
## Q0045                                   Cytochrome C and Quinol oxidase polypeptide I
## Q0050    Reverse transcriptase (RNA-dependent DNA polymerase);Type II intron maturase
## Q0055    Reverse transcriptase (RNA-dependent DNA polymerase);Type II intron maturase
## Q0060            Cytochrome C and Quinol oxidase polypeptide I;LAGLIDADG endonuclease
## Q0065            Cytochrome C and Quinol oxidase polypeptide I;LAGLIDADG endonuclease
## Q0070 Cytochrome C and Quinol oxidase polypeptide I;LAGLIDADG DNA endonuclease family
##         annot_pfam_id annot_pirsf_description annot_pirsf_id
## Q0045         PF00115                     N/A            N/A
## Q0050 PF00078;PF01348                     N/A            N/A
## Q0055 PF00078;PF01348                     N/A            N/A
## Q0060 PF00115;PF00961                     N/A            N/A
## Q0065 PF00115;PF00961                     N/A            N/A
## Q0070 PF00115;PF03161                     N/A            N/A
##                                                                                                            annot_predicted_go_component
## Q0045 integral component of membrane;membrane;mitochondrial inner membrane;mitochondrion;respiratory chain;respiratory chain complex IV
## Q0050                                                                                                                     mitochondrion
## Q0055                                                                                      integral component of membrane;mitochondrion
## Q0060                                                                             integral component of membrane;membrane;mitochondrion
## Q0065                                                                                      integral component of membrane;mitochondrion
## Q0070                                                                             integral component of membrane;membrane;mitochondrion
##                                                                                 annot_predicted_go_function
## Q0045                  cytochrome-c oxidase activity;heme binding;metal ion binding;oxidoreductase activity
## Q0050                                                                                                      
## Q0055                                                            cytochrome-c oxidase activity;heme binding
## Q0060 cytochrome-c oxidase activity;endonuclease activity;heme binding;hydrolase activity;nuclease activity
## Q0065 cytochrome-c oxidase activity;endonuclease activity;heme binding;hydrolase activity;nuclease activity
## Q0070 cytochrome-c oxidase activity;endonuclease activity;heme binding;hydrolase activity;nuclease activity
##                                         annot_predicted_go_id_component
## Q0045 GO:0016021;GO:0016020;GO:0005743;GO:0005739;GO:0070469;GO:0045277
## Q0050                                                        GO:0005739
## Q0055                                             GO:0016021;GO:0005739
## Q0060                                  GO:0016021;GO:0016020;GO:0005739
## Q0065                                             GO:0016021;GO:0005739
## Q0070                                  GO:0016021;GO:0016020;GO:0005739
##                               annot_predicted_go_id_function
## Q0045            GO:0004129;GO:0020037;GO:0046872;GO:0016491
## Q0050                                                       
## Q0055                                  GO:0004129;GO:0020037
## Q0060 GO:0004129;GO:0004519;GO:0020037;GO:0016787;GO:0004518
## Q0065 GO:0004129;GO:0004519;GO:0020037;GO:0016787;GO:0004518
## Q0070 GO:0004129;GO:0004519;GO:0020037;GO:0016787;GO:0004518
##                                           annot_predicted_go_id_process
## Q0045                                  GO:0009060;GO:0055114;GO:0006119
## Q0050                                             GO:0006397;GO:0055114
## Q0055 GO:0006278;GO:0009060;GO:0022900;GO:0006397;GO:0055114;GO:1902600
## Q0060 GO:0008380;GO:0009060;GO:0006314;GO:0006397;GO:0090305;GO:0055114
## Q0065 GO:0008380;GO:0009060;GO:0006314;GO:0006397;GO:0090305;GO:0055114
## Q0070 GO:0008380;GO:0009060;GO:0006314;GO:0006397;GO:0090305;GO:0055114
##                                                                                                                                           annot_predicted_go_process
## Q0045                                                                                      aerobic respiration;oxidation-reduction process;oxidative phosphorylation
## Q0050                                                                                                                    mRNA processing;oxidation-reduction process
## Q0055 RNA-dependent DNA biosynthetic process;aerobic respiration;electron transport chain;mRNA processing;oxidation-reduction process;proton transmembrane transport
## Q0060                         RNA splicing;aerobic respiration;intron homing;mRNA processing;nucleic acid phosphodiester bond hydrolysis;oxidation-reduction process
## Q0065                         RNA splicing;aerobic respiration;intron homing;mRNA processing;nucleic acid phosphodiester bond hydrolysis;oxidation-reduction process
## Q0070                         RNA splicing;aerobic respiration;intron homing;mRNA processing;nucleic acid phosphodiester bond hydrolysis;oxidation-reduction process
##       annot_project_id
## Q0045          FungiDB
## Q0050          FungiDB
## Q0055          FungiDB
## Q0060          FungiDB
## Q0065          FungiDB
## Q0070          FungiDB
##                                                               annot_prositeprofiles_description
## Q0045                                                     Cytochrome oxidase subunit I  profile
## Q0050                                       Reverse transcriptase (RT) catalytic domain profile
## Q0055 Cytochrome oxidase subunit I  profile;Reverse transcriptase (RT) catalytic domain profile
## Q0060                                                     Cytochrome oxidase subunit I  profile
## Q0065                                                     Cytochrome oxidase subunit I  profile
## Q0070                                                     Cytochrome oxidase subunit I  profile
##       annot_prositeprofiles_id annot_protein_length
## Q0045                  PS50855                  534
## Q0050                  PS50878                  834
## Q0055          PS50855;PS50878                  854
## Q0060                  PS50855                  415
## Q0065                  PS50855                  556
## Q0070                  PS50855                  630
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       annot_protein_sequence
## Q0045                                                                                                                                                                                                                                                                                                                                 MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNVLVVGHAVLMIFFLVMPALIGGFGNYLLPLMIGATDTAFPRINNIAFWVLPMGLVCLVTSTLVESGAGTGWTVYPPLSSIQAHSGPSVDLAIFALHLTSISSLLGAINFIVTTLNMRTNGMTMHKLPLFVWSIFITAFLLLLSLPVLSAGITMLLLDRNFNTSFFEVSGGGDPILYEHLFWFFGHPEVYILIIPGFGIISHVVSTYSKKPVFGEISMVYAMASIGLLGFLVWSHHMYIVGLDADTRAYFTSATMIIAIPTGIKIFSWLATIHGGSIRLATPMLYAIAFLFLFTMGGLTGVALANASLDVAFHDTYYVVGHFHYVLSMGAIFSLFAGYYYWSPQILGLNYNEKLAQIQFWLIFIGANVIFFPMHFLGINGMPRRIPDYPDAFAGWNYVASIGSFIATLSLFLFIYILYDQLVNGLNNKVNNKSVIYNKAPDFVESNTIFNLNTVKSSSIEFLLTSPPAVHSFNTPAVQS
## Q0050                     MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNGAPTSAYISLMRTALVLWIINRYLKHMTNSVGANFTGTMACHKTPMISVGGVKCYMVRLTNFLQVFIRITISSYHLDMVKQVWLFYVEVIRLWFIVLDSTGSVKKMKDTNNTKGNTKSEGSTERGNSGVDRGMVVPNTQMKMRFLNQVRYYSVNNNLKMGKDTNIELSKDTSTSDLLEFEKLVMDNMNEENMNNNLLSIMKNVDMLMLAYNRIKSKPGNMTPGTTLETLDGMNMMYLNKLSNELGTGKFKFKPMRMVNIPKPKGGMRPLSVGNPRDKIVQEVMRMILDTIFDKKMSTHSHGFRKNMSCQTAIWEVRNMFGGSNWFIEVDLKKCFDTISHDLIIKELKRYISDKGFIDLVYKLLRAGYIDEKGTYHKPMLGLPQGSLISPILCNIVMTLVDNWLEDYINLYNKGKVKKQHPTYKKLSRMIAKAKMFSTRLKLHKERAKGPTFIYNDPNFKRMKYVRYADDILIGVLGSKNDCKMIKRDLNNFLNSLGLTMNEEKTLITCATETPARFLGYNISITPLKRMPTVTKTIRGKTIRSRNTTRPIINAPIRDIINKLATNGYCKHNKNGRMGVPTRVGRWTYEEPRTIINNYKALGRGILNYYKLATNYKRLRERIYYVLYYSCVLTLASKYRLKTMSKTIKKFGYNLNIIENDKLIANFPRNTFDNIKKIENHGMFMYMSEAKVTDPFEYIDSIKYMLPTAKANFNKPCSICNSTIDVEMHHVKQLHRGMLKATKDYITGRMITMNRKQIPLCKQCHIKTHKNKFKNMGPGM
## Q0055 MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNVLVVGHAVLMIFCAPFRLIYHCIEVLIDKHISVYSINENFTVSFWFWLLVVTYMVFRYVNHMAYPVGANSTGTMACHKSAGVKQPAQGKNCPMARLTNSCKECLGFSLTPSHLGIVIHAYVLEEEVHELTKNESLALSKSWHLEGCTSSNGKLRNTGLSERGNPGDNGVFMVPKFNLNKVRYFSTLSKLNARKEDSLAYLTKINTTDFSELNKLMENNHNKTETINTRILKLMSDIRMLLIAYNKIKSKKGNMSKGSNNITLDGINISYLNKLSKDINTNMFKFSPVRRVEIPKTSGGFRPLSVGNPREKIVQESMRMMLEIIYNNSFSYYSHGFRPNLSCLTAIIQCKNYMQYCNWFIKVDLNKCFDTIPHNMLINVLNERIKDKGFMDLLYKLLRAGYVDKNNNYHNTTLGIPQGSVVSPILCNIFLDKLDKYLENKFENEFNTGNMSNRGRNPIYNSLSSKIYRCKLLSEKLKLIRLRDHYQRNMGSDKSFKRAYFVRYADDIIIGVMGSHNDCKNILNDINNFLKENLGMSINMDKSVIKHSKEGVSFLGYDVKVTPWEKRPYRMIKKGDNFIRVRHHTSLVVNAPIRSIVMKLNKHGYCSHGILGKPRGVGRLIHEEMKTILMHYLAVGRGIMNYYRLATNFTTLRGRITYILFYSCCLTLARKFKLNTVKKVILKFGKVLVDPHSKVSFSIDDFKIRHKMNMTDSNYTPDEILDRYKYMLPRSLSLFSGICQICGSKHDLEVHHVRTLNNAANKIKDDYLLGRMIKMNRKQITICKTCHFKVHQGKYNGPGL
## Q0060                                                                                                                                                                                                                                                                                                                                                                                                                                                        MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNVLVVGHAVLMIFFLVMPALIGGFGNQKRYESNNNNNQVMENKEYNLKLNYDKLGPYLAGLIEGDGTITVQNSSSMKKSKYRPLIVVVFKLEDLELANYLCNLTKCGKVYKKINRNYVLWTIHDLKGVYTLLNIINGYMRTPKYEAFVRGAEFMNNYINSTTITHNKLKNMDNIKIKPLDTSDIGSNAWLAGMTDADGNFSINLMNGKNRSSRAMPYYCLELRQNYQKNSNNNNINFSYFYIMSAIATYFNVNLYSRERNLNLLVSTNNTYKTYYSYKVMVANTYKNIKVMEYFNKYSLLSSKHLDFLDWSKLVILINNEGQSMKTNGSWELGMNLRKDYNKTRTTFTWSHLKNTYLENK
## Q0065                                                                                                                                                                                                                                                                                                           MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNVLVVGHAVLMIFFLVMPALIGGFGNYLLPLMIGATDTAFPRINNIAFWVLPMGLVCLVTSTLVESGAGTGWTVYPPLSSIQAHSGPSVDLAIFALHLTSISSLLGAINFIVTTLNMRTNGMTMHKLPLFVWSIFITAFLLLLSLPVLSAGITMLLLDRNFNTSFFEVSGGGDPILYEHLFWFFGQTVATIIMLMMYNDMHFSKCWKLLKKWITNIMSTLFKALFVKMFMSYNNQQDKMMNNTMLKKDNIKRSSETTRKMLNNSMNKKFNQWLAGLIDGDGYFGIVSKKYVSLEITVALEDEMALKEIQNKFGGSIKLRSGVKAIRYRLTNKTGMIKLINAVNGNIRNTKRLVQFNKVCILLGIDFIYPIKLTKDNSWFVGFFDADGTINYSFKNNHPQLTISVTNKYLQDVQEYKNILGGNIYFDKSQNGYYKWSIQSKDMVLNFINDYIKMNPSRTTKMNKLYLSKEFYNLKELKAYNKSSDSMQYKAWLNFENKWKNK
## Q0070                                                                                                                                                                                                                                 MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSLIIRLELAAPGSQYLHGNSQLFNVLVVGHAVLMIFFLVMPALIGGFGNYLLPLMIGATDTAFPRINNIAFWVLPMGLVCLVTSTLVESGAGTGWTVYPPLSSIQAHSGPSVDLAIFALHLTSISSLLGAINFIVTTLNMRTNGMTMHKLPLFVWSIFITAFLLLLSLPVLSAGITMLLLDRNFNTSFFEVSGGGDPILYEHLFWFFGHPEVYILIIPGFGIISHVVSTYSKKPVFGEISMVYAMASIGLLGFLVWSHHMYIVGLDADTRAYFTSATMIIAIPTGIKIFSWLMNPFSKDKNKNKNKKLIRNYQKMNNNNMMKTYLNNNNMIMMNMYKGNLYDIYPRSNRNYIQPNNINKELVVYGYNLESCVGMPTYTNIVKHMVGIPNNILYIMTGILLTDGWIDYTSKKDLDKKTIMEINCRFRLKQSMIHSEYLMYVFMLLSHYCMSYPKMKIAKVKGKSYNQLEFYTRSLPCFTILRYMFYNGRVKIVPNNLYDLLNYESLAHMIMCDGSFVKGGGLYLNLQSFTTKELIFIMNILKIKFNLNCTLHKSRNKYTIYMRVESVKRLFPMIYKYILPSMRYKFDIMLWQKKYNMIN
##       annot_sequence_id
## Q0045          KP263414
## Q0050          KP263414
## Q0055          KP263414
## Q0060          KP263414
## Q0065          KP263414
## Q0070          KP263414
##                                                                 annot_signalp_peptide
## Q0045 NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
## Q0050 HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
## Q0055 HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
## Q0060 HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
## Q0065 HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
## Q0070 HMM: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL, NN: MVQRWLYSTNAKDIAVLYFMLAIFSGMAGTAMSL
##                      annot_signalp_scores annot_smart_description
## Q0045 NN Sum: 3, NN D: .54, HMM Prob: .72                     N/A
## Q0050 NN Sum: 3, NN D: .53, HMM Prob: .72                        
## Q0055 NN Sum: 3, NN D: .54, HMM Prob: .72                        
## Q0060 NN Sum: 3, NN D: .54, HMM Prob: .72                     N/A
## Q0065 NN Sum: 3, NN D: .54, HMM Prob: .72                     N/A
## Q0070 NN Sum: 3, NN D: .54, HMM Prob: .72                     N/A
##       annot_smart_id annot_source_id annot_strand
## Q0045            N/A     Q0045-t26_1      forward
## Q0050        SM00507     Q0050-t26_1      forward
## Q0055        SM00507     Q0055-t26_1      forward
## Q0060            N/A     Q0060-t26_1      forward
## Q0065            N/A     Q0065-t26_1      forward
## Q0070            N/A     Q0070-t26_1      forward
##                                  annot_superfamily_description
## Q0045                      Cytochrome c oxidase subunit I-like
## Q0050  DNA/RNA polymerases;Cytochrome c oxidase subunit I-like
## Q0055  DNA/RNA polymerases;Cytochrome c oxidase subunit I-like
## Q0060 Homing endonucleases;Cytochrome c oxidase subunit I-like
## Q0065 Homing endonucleases;Cytochrome c oxidase subunit I-like
## Q0070 Homing endonucleases;Cytochrome c oxidase subunit I-like
##       annot_superfamily_id annot_three_prime_utr_length
## Q0045             SSF81442                             
## Q0050    SSF56672;SSF81442                             
## Q0055    SSF56672;SSF81442                             
## Q0060    SSF55608;SSF81442                             
## Q0065    SSF55608;SSF81442                             
## Q0070    SSF55608;SSF81442                             
##       annot_tigrfam_description annot_tigrfam_id annot_tm_count
## Q0045                       N/A              N/A             12
## Q0050                       N/A              N/A              1
## Q0055                       N/A              N/A              3
## Q0060                       N/A              N/A              3
## Q0065                       N/A              N/A              6
## Q0070                       N/A              N/A             10
##       annot_trans_found_per_gene_internal annot_transcript_index_per_gene
## Q0045                                   1                               1
## Q0050                                   1                               1
## Q0055                                   1                               1
## Q0060                                   1                               1
## Q0065                                   1                               1
## Q0070                                   1                               1
##       annot_transcript_length annot_transcript_link
## Q0045                  1605.0           Q0045-t26_1
## Q0050                  2505.0           Q0050-t26_1
## Q0055                  2565.0           Q0055-t26_1
## Q0060                  1248.0           Q0060-t26_1
## Q0065                  1671.0           Q0065-t26_1
## Q0070                  1893.0           Q0070-t26_1
##                        annot_transcript_product
## Q0045            cytochrome c oxidase subunit 1
## Q0050  intron-encoded reverse transcriptase aI1
## Q0055  intron-encoded reverse transcriptase aI2
## Q0060       intron-encoded DNA endonuclease aI3
## Q0065       intron-encoded DNA endonuclease aI4
## Q0070 intron-encoded DNA endonuclease aI5 alpha
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   annot_transcript_sequence
## Q0045                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCACCCTGAAGTATATATTTTAATTATTCCTGGATTTGGTATTATTTCACATGTAGTATCAACATATTCTAAAAAACCTGTATTTGGTGAAATTTCAATGGTATATGCTATGGCTTCAATTGGATTATTAGGATTCTTAGTATGATCACATCATATGTATATTGTAGGATTAGATGCAGATCTTAGAGCATATTTCCTATCTGCACTAATGATTATTGCAATTCCAACAGGAATTAAAATTTTCTCATGATTAGCTCTAATCCATGGTGGTTCAATTAGATTAGCACTACCTATGTTATATGCAATTGCATTCTTATTCTTATTCACAATGGGTGGTTTAACTGGTGTTGCCTTAGCTAACGCCTCATTAGATGTAGCATTCCACGATACTTACTACGTGGTGGGACATTTTCACTATGTATTATCAATGGGTGCTATTTTCTCTTTATTTGCAGGATACTATTATTGAAGTCCTCAAATTTTAGGTTTAAACTATAATGAAAAATTAGCTCAAATTCAATTCTGATTAATTTTCATTGGGGCTAATGTTATTTTCTTCCCAATGCATTTTTTAGGTATTAATGGTATGCCTAGAAGAATTCCTGATTATCCTGATGCTTTCGCAGGATGAAATTATGTCGCTTCTATTGGTTCATTCATTGCACTATTATCATTATTCTTATTTATCTATATTTTATATGATCAATTAGTTAATGGATTAAACAATAAAGTTAATAATAAATCAGTTATTTATAATAAAGCACCTGATTTTGTAGAATCTAATCTTATCTTTAATTTAAATACAGTTAAATCTTCATCTATCGAATTCTTATTAACTTCTCCACCAGCTGTACACTCATTTAATACACCAGCTGTACAATCTTAA
## Q0050                                                             ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGGTGCGCCTCTCAGTGCGTATATTTCGTTGATGCGTCTAGCATTAGTATTATGAATCATCAATAGATACTTAAAACATATGACTAACTCAGTAGGGGCTAACTTTACGGGGACAATAGCATGTCATAAAACACCTATGATTAGTGTAGGTGGAGTTAAGTGTTACATGGTTAGGTTAACGAACTTCTTACAAGTCTTTATCAGGATTACAATTTCCTCTTATCATTTGGATATAGTAAAACAAGTTTGATTATTTTACGTTGAGGTAATCAGATTATGATTCATTGTTTTAGATAGCACAGGCAGTGTGAAAAAGATGAAGGACCTAAATAACACAAAAGGAAATACGAAAAGTGAGGGATCAACTGAAAGAGGAAACTCTGGAGTTGACAGAGGTATAGTAGTACCGAATACTCAAATAAAAATGAGATTTTTAAATCAAGTTAGATACTATTCAGTAAATAATAATTTAAAAATAGGGAAGGATACCAATATTGAGTTATCAAAAGATACAAGTACTTCGGACTTGTTAGAATTTGAGAAATTAGTAATAGATAATATAAATGAGGAAAATATAAATAATAATTTATTAAGTATTATAAAAAACGTAGATATATTAATATTAGCATATAATAGAATTAAGAGTAAACCTGGTAATATAACTCCAGGTACAACATTAGAAACATTAGATGGTATAAATATAATATATTTAAATAAATTATCAAATGAATTAGGAACAGGTAAATTCAAATTTAAACCCATGAGAATAGTTAATATTCCTAAACCTAAAGGTGGTATAAGACCTTTAAGTGTAGGTAATCCAAGAGATAAAATTGTACAAGAAGTTATAAGAATAATTTTAGATACAATTTTTGATAAAAAGATATCAACACATTCACATGGTTTTAGAAAGAATATAAGTTGTCAAACAGCAATTTGAGAAGTTAGAAATATATTTGGTGGAAGTAATTGATTTATTGAAGTAGACTTAAAAAAATGTTTTGATACAATTTCTCATGATTTAATTATTAAAGAATTAAAAAGATATATTTCAGATAAAGGTTTTATTGATTTAGTATATAAATTATTAAGAGCTGGTTATATTGATGAGAAAGGAACTTATCATAAACCTATATTAGGTTTACCTCAAGGATCATTAATTAGTCCTATCTTATGTAATATTGTAATAACATTGGTAGATAATTGATTAGAAGATTATATTAATTTATATAATAAAGGTAAAGTTAAAAAACAACATCCTACATATAAAAAATTATCAAGAATAATTGCAAAAGCTAAAATATTTTCGACAAGATTAAAATTACATAAAGAAAGAGCTAAAGGCCCACTATTTATTTATAATGATCCTAATTTCAAGAGAATAAAATACGTTAGATATGCAGATGATATTTTAATTGGGGTATTAGGTTCAAAAAATGATTGTAAAATAATCAAAAGAGATTTAAACAATTTTTTAAATTCATTAGGTTTAACTATAAATGAAGAAAAAACTTTAATTACTTGTGCAACTGAACTACCAGCAAGATTTTTAGGTTATAATATTTCAATTACACCTTTAAAAAGAATACCTACAGTTACTAAACTAATTAGAGGTAAACTTATTAGAAGTAGAAATACAACTAGACCTATTATTAATGCACCAATTAGAGATATTATCAATAAATTAGCTACTAATGGATATTGTAAGCATAATAAAAATGGTAGAATAGGAGTGCCTACAAGAGTAGGTAGATGACTATATGAAGAACCTAGAACAATTATTAATAATTATAAAGCGTTAGGTAGAGGTATCTTAAATTATTATAAATTAGCTACTAATTATAAAAGATTAAGAGAAAGAATCTATTACGTATTATATTATTCATGTGTATTAACTTTAGCTAGTAAATATAGATTAAAAACAATAAGTAAAACTATTAAAAAATTTGGTTATAATTTAAATATTATTGAAAATGATAAATTAATTGCCAATTTTCCAAGAAATACTTTTGATAATATCAAAAAAATTGAAAATCATGGTATATTTATATATATATCAGAAGCTAAAGTAACTGATCCTTTTGAATATATCGATTCAATTAAATATATATTACCTACAGCTAAAGCTAATTTTAATAAACCTTGTAGTATTTGTAATTCAACTATTGATGTAGAAATACATCATGTTAAACAATTACATAGAGGTATATTAAAAGCACTTAAAGATTATATTCTAGGTAGAATAATTACCATAAACAGAAAACAAATTCCATTATGTAAACAATGTCATATTAAAACACATAAAAATAAATTTAAAAATATAGGACCTGGTATATAA
## Q0055 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTGTGCGCCGTTTCGCTTAATTTATCACTGTATTGAAGTGTTAATTGATAAACATATCTCTGTTTATTCAATTAATGAAAACTTTACCGTATCATTTTGGTTCTGATTATTAGTAGTAACATACATAGTATTTAGATACGTAAACCATATGGCTTACCCAGTTGGGGCCAACTCAACGGGGACAATAGCATGCCATAAAAGCGCTGGAGTAAAACAGCCAGCGCAAGGTAAGAACTGTCCGATGGCTAGGTTAACGAATTCCTGTAAAGAATGTTTAGGGTTCTCATTAACTCCTTCCCACTTGGGGATTGTGATTCATGCTTATGTATTGGAAGAAGAGGTACACGAGTTAACCAAAAATGAATCATTAGCTTTAAGTAAAAGTTGACATTTGGAGGGCTGTACGAGTTCAAATGGAAAATTAAGAAATACGGGATTGTCCGAAAGGGGAAACCCTGGGGATAACGGAGTCTTCATAGTACCCAAATTTAATTTAAATAAAGTGAGATACTTTAGTACTTTATCTAAATTAAATGCAAGGAAGGAAGACAGTTTAGCGTATTTAACAAAGATTAATACTACGGATTTTTCCGAGTTAAATAAATTAATAGAAAATAATCATAATAAACTTGAAACCATTAATACTAGAATTTTAAAATTAATGTCAGATATTAGAATGTTATTAATTGCTTATAATAAAATTAAAAGTAAGAAAGGTAATATATCTAAAGGTTCTAATAATATTACCTTAGATGGGATTAATATTTCATATTTAAATAAATTATCTAAAGATATTAACACTAATATGTTTAAATTTTCTCCGGTTAGAAGAGTTGAAATTCCTAAAACATCTGGAGGATTTAGACCTTTAAGTGTTGGAAATCCTAGAGAAAAAATTGTACAAGAAAGTATGAGAATAATATTAGAAATTATCTATAATAATAGTTTCTCTTATTATTCTCATGGATTTAGACCTAACTTATCTTGTTTAACAGCTATTATTCAATGTAAAAATTATATGCAATACTGTAATTGATTTATTAAAGTAGATTTAAATAAATGCTTTGATACAATTCCACATAATATGTTAATTAATGTATTAAATGAGAGAATCAAAGATAAAGGTTTCATAGACTTATTATATAAATTATTAAGAGCTGGATATGTTGATAAAAATAATAATTATCATAATACAACTTTAGGAATTCCTCAAGGTAGTGTTGTCAGTCCTATTTTATGTAATATTTTTTTAGATAAATTAGATAAATATTTAGAAAATAAATTTGAGAATGAATTCAATACTGGAAATATGTCTAATAGAGGTAGAAATCCAATTTATAATAGTTTATCATCTAAAATTTATAGATGTAAATTATTATCTGAAAAATTAAAATTGATTAGATTAAGAGACCATTACCAAAGAAATATGGGATCTGATAAAAGTTTTAAAAGAGCTTATTTTGTTAGATATGCTGATGATATTATCATTGGTGTAATGGGTTCTCATAATGATTGTAAAAATATTTTAAACGATATTAATAACTTCTTAAAAGAAAATTTAGGTATGTCAATTAATATAGATAAATCCGTTATTAAACATTCTAAAGAAGGAGTTAGTTTTTTAGGGTATGATGTAAAAGTTACACCTTGAGAAAAAAGACCTTATAGAATGATTAAAAAAGGTGATAATTTTATTAGGGTTAGACATCATACTAGTTTAGTTGTTAATGCCCCTATTAGAAGTATTGTAATAAAATTAAATAAACATGGCTATTGTTCTCATGGTATTTTAGGAAAACCCAGAGGGGTTGGAAGATTAATTCATGAAGAAATGAAAACCATTTTAATGCATTACTTAGCTGTTGGTAGAGGTATTATAAACTATTATAGATTAGCTACCAATTTTACCACATTAAGAGGTAGAATTACATACATTTTATTTTATTCATGTTGTTTAACATTAGCAAGAAAATTTAAATTAAATACTGTTAAGAAAGTTATTTTAAAATTCGGTAAAGTATTAGTTGATCCTCATTCAAAAGTTAGTTTTAGTATTGATGATTTTAAAATTAGACATAAAATAAATATAACTGATTCTAATTATACACCTGATGAAATTTTAGATAGATATAAATATATGTTACCTAGATCTTTATCATTATTTAGTGGTATTTGTCAAATTTGTGGTTCTAAACATGATTTAGAAGTACATCACGTAAGAACATTAAATAATGCTGCCAATAAAATTAAAGATGATTATTTATTAGGTAGAATGATTAAGATAAATAGAAAACAAATTACTATCTGTAAAACATGTCATTTTAAAGTTCATCAAGGTAAATATAATGGTCCAGGTTTATAA
## Q0060                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACCAAAAAAGATATGAAAGTAATAATAATAATAATCAAGTAATAGAAAATAAAGAATATAATTTAAAATTAAATTATGATAAGTTGGGACCTTATTTAGCTGGATTAATTGAAGGTGATGGAACTATTCTAGTTCAAAATTCATCTTCAATAAAAAAATCTAAATATAGACCGTTAATTGTTGTAGTATTTAAATTAGAAGATTTAGAATTAGCTAATTATTTATGTAATTTAACTAAATGTGGAAAAGTGTATAAAAAAATTAATCGTAATTATGTATTATGACTTATTCATGATTTAAAAGGTGTATATACATTATTAAATATTATTAATGGATATATGAGAACACCTAAATATGAAGCATTTGTTAGAGGTGCTGAATTTATAAATAATTATATTAATTCAACAACAATTCTACATAATAAATTAAAAAATATAGATAATATTAAAATTAAACCATTAGATACATCAGATATTGGTTCAAACGCTTGATTAGCTGGTATGACAGATGCAGATGGTAATTTTTCTATTAATTTAATAAATGGTAAAAATCGTTCTAGTAGAGCAATGCCTTATTATTGTTTAGAATTAAGACAAAATTATCAAAAAAATTCTAATAATAATAATATTAATTTTTCTTATTTTTATATTATGTCTGCAATTGCACTATATTTTAATGTTAATTTATATAGTAGAGAACGTAATTTAAATTTATTAGTATCTCTTAATAATACGTATAAACTATATTATAGTTATAAAGTAATAGTGGCTAATCTATATAAAAATATTAAAGTAATAGAATACTTTAATAAATATTCTTTATTATCATCTAAACACTTAGATTTTTTAGATTGATCTAAATTAGTTATTTTAATTAATAATGAGGGTCAAAGTATAAAACTTAATGGTAGTTGAGAATTAGGTATAAATTTACGTAAAGATTATAATAAAACTAGAACTACGTTTACTTGATCTCATTTAAAAAATACATATTTAGAAAATAAATAA
## Q0065                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCAAACAGTGGCCCTTATTATTATATTAATAATATATAATGATATGCATTTTTCTAAATGCTGGAAATTATTAAAAAAATGAATTACAAATATTATAAGTCTATTATTTAAAGCCTTATTTGTAAAAATATTCATATCTTATAATAATCAGCAGGATAAGATAATAAATAATCTTATATTAAAAAAAGATAATATTAAAAGATCCTCAGAGACTACAAGAAAAATATTAAATAATTCAATAAATAAAAAATTTAATCAATGATTAGCTGGATTAATTGATGGTGATGGATATTTTGGTATTGTAAGTAAGAAATATGTATCATTAGAAATTCTAGTAGCATTAGAAGATGAAATAGCTTTAAAAGAAATTCAAAATAAATTTGGTGGTTCTATTAAATTAAGATCAGGTGTAAAAGCTATTAGATATAGATTACTTAATAAAACTGGTATAATTAAATTAATTAATGCAGTTAATGGTAATATTAGAAATACTAAAAGATTAGTACAATTTAATAAAGTTTGTATTTTATTAGGTATTGATTTTATTTATCCAATTAAATTAACTAAAGATAATAGTTGATTTGTTGGATTTTTTGATGCTGATGGTACAATTAATTATTCATTTAAAAATAATCATCCTCAATTAACAATTTCTGTAACTAATAAATATTTACAAGATGTACAAGAATATAAAAATATTTTAGGTGGTAATATTTATTTTGATAAATCACAAAATGGTTATTATAAATGATCCATTCAATCAAAAGATATAGTATTAAATTTTATTAATGATTATATTAAAATAAATCCATCAAGAACACTAAAAATAAATAAATTATATTTAAGTAAAGAATTTTATAATTTAAAAGAATTAAAAGCTTATAATAAATCTTCTGATTCAATACAATATAAAGCATGATTAAATTTTGAAAATAAATGAAAAAATAAATAA
## Q0070                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 ATGGTACAAAGATGATTATATTCAACAAATGCAAAAGATATTGCAGTATTATATTTTATGTTAGCTATTTTTAGTGGTATGGCAGGAACAGCAATGTCTTTAATCATTAGATTAGAATTAGCTGCACCTGGTTCACAATATTTACATGGTAATTCACAATTATTTAATGTTTTAGTAGTTGGTCATGCTGTATTAATGATTTTCTTCTTAGTAATGCCTGCTTTAATTGGAGGTTTTGGTAACTATTTATTACCATTAATAATTGGAGCTACAGATACAGCATTTCCAAGAATTAATAACATTGCTTTTTGAGTATTACCTATGGGGTTAGTATGTTTAGTTACATCAACTTTAGTAGAATCAGGTGCTGGTACAGGGTGAACTGTCTATCCACCATTATCATCTATTCAGGCACATTCAGGACCTAGTGTAGATTTAGCAATTTTTGCATTACATTTAACATCAATTTCATCATTATTAGGTGCTATTAATTTCATTGTAACAACATTAAATATGAGAACAAATGGTATGACAATGCATAAATTACCATTATTTGTATGATCAATTTTCATTACAGCGTTCTTATTATTATTATCATTACCTGTATTATCTGCTGGTATTACAATGTTATTATTAGATAGAAACTTCAATACTTCATTCTTTGAAGTATCAGGAGGTGGTGACCCAATCTTATACGAGCATTTATTTTGATTCTTTGGTCACCCTGAAGTATATATTTTAATTATTCCTGGATTTGGTATTATTTCACATGTAGTATCAACATATTCTAAAAAACCTGTATTTGGTGAAATTTCAATGGTATATGCTATGGCTTCAATTGGATTATTAGGATTCTTAGTATGATCACATCATATGTATATTGTAGGATTAGATGCAGATCTTAGAGCATATTTCCTATCTGCACTAATGATTATTGCAATTCCAACAGGAATTAAAATTTTCTCATGATTAATAAATCCCTTTAGCAAGGATAAAAATAAAAATAAAAATAAAAAGTTGATCAGAAATTATCAAAAAATAAATAATAATAATATAATAAAAACATATTTAAATAATAATAATATAATTATAATAAATATATATAAAGGTAATTTATATGATATTTATCCAAGATCAAATAGAAATTATATTCAACCAAATAATATTAATAAAGAATTAGTAGTATATGGTTATAATTTAGAATCTTGTGTTGGTATACCTCTATATACTAATATTGTAAAACATATAGTAGGTATTCCTAATAATATTTTATATATTATAACAGGTATTTTATTAACAGATGGTTGAATTGATTATCTATCTAAAAAAGATTTAGATAAAAAAACAATTATAGAAATTAATTGTAGATTTAGATTAAAACAATCAATAATTCATAGTGAATATTTAATATATGTATTTATATTATTATCACATTATTGTATAAGTTATCCTAAAATAAAAATTGCTAAAGTTAAAGGTAAATCATATAATCAATTAGAATTTTATACTAGATCATTACCATGTTTTACTATTTTAAGATATATATTTTATAATGGTAGAGTAAAAATTGTACCTAATAATTTATATGATTTATTAAATTATGAATCTTTAGCTCATATAATTATATGTGATGGTTCATTTGTAAAAGGTGGAGGTTTATATTTAAATTTACAATCTTTTCTAACTAAAGAATTAATTTTTATTATAAATATTTTAAAAATTAAATTTAATTTAAATTGTCTATTACATAAATCTAGAAATAAATATCTTATTTATATAAGAGTAGAATCTGTTAAAAGATTATTTCCTATAATTTATAAATATATTTTACCTTCTATAAGATATAAATTTGATATTATATTATGACAAAAAAAATATAATATGATTAATTAA
##       annot_transcripts_found_per_gene annot_uniprot_id annot_wdk_weight
## Q0045                           1 of 1           P00401                0
## Q0050                           1 of 1           P03875                0
## Q0055                           1 of 1           P03876                0
## Q0060                           1 of 1           P03877                0
## Q0065                           1 of 1           P03878                0
## Q0070                           1 of 1           Q9ZZX1                0</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb123-1" title="1"><span class="co">## Yay! We have data about S. cerevisiae!</span></a>
<a class="sourceLine" id="cb123-2" title="2"></a>
<a class="sourceLine" id="cb123-3" title="3">chosen_columns_idx &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="dt">x=</span>avail_columns, <span class="dt">pattern=</span><span class="st">&quot;^GO&quot;</span>)</a>
<a class="sourceLine" id="cb123-4" title="4">chosen_columns &lt;-<span class="st"> </span>avail_columns[chosen_columns_idx]</a>
<a class="sourceLine" id="cb123-5" title="5">sc_go &lt;-<span class="st"> </span><span class="kw">load_orgdb_go</span>(sc_orgdb, <span class="dt">columns=</span>chosen_columns)</a></code></pre></div>
<pre><code>## Loading required namespace: GO.db</code></pre>
<pre><code>## </code></pre>
<pre><code>## This is an orgdb, good.</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb128-1" title="1"><span class="kw">head</span>(sc_go)</a></code></pre></div>
<pre><code>## # A tibble: 6 x 11
##   GID   GO_EVIDENCE_CODE GO_GO_ID GO_GO_TERM_NAME GO_IS_NOT GO_ONTOLOGY
##   &lt;chr&gt; &lt;chr&gt;            &lt;chr&gt;    &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;      
## 1 Q0045 IEA              GO:0006… oxidative phos… N/A       Biological…
## 2 Q0045 IBA              GO:0006… mitochondrial … N/A       Biological…
## 3 Q0045 IDA              GO:0006… mitochondrial … N/A       Biological…
## 4 Q0045 IBA              GO:0009… aerobic respir… N/A       Biological…
## 5 Q0045 IEA              GO:0009… aerobic respir… N/A       Biological…
## 6 Q0045 IEA              GO:0009… aerobic respir… N/A       Biological…
## # … with 5 more variables: GO_REFERENCE &lt;chr&gt;, GO_SORT_KEY &lt;chr&gt;,
## #   GO_SOURCE &lt;chr&gt;, GO_SUPPORT_FOR_EVIDENCE_CODE_ASSIGNMENT &lt;chr&gt;,
## #   GO_TRANSCRIPT_ID_S &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb130-1" title="1"><span class="co">## Yay Gene ontology data for Crithidia!</span></a>
<a class="sourceLine" id="cb130-2" title="2"></a>
<a class="sourceLine" id="cb130-3" title="3">chosen_columns_idx &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="dt">x=</span>avail_columns, <span class="dt">pattern=</span><span class="st">&quot;^INTERPRO&quot;</span>)</a>
<a class="sourceLine" id="cb130-4" title="4">chosen_columns &lt;-<span class="st"> </span>avail_columns[chosen_columns_idx]</a>
<a class="sourceLine" id="cb130-5" title="5">sc_interpro &lt;-<span class="st"> </span><span class="kw">load_orgdb_go</span>(sc_orgdb, <span class="dt">columns=</span>chosen_columns)</a></code></pre></div>
<pre><code>## This is an orgdb, good.
## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb132-1" title="1"><span class="kw">head</span>(sc_interpro)</a></code></pre></div>
<pre><code>## # A tibble: 6 x 10
##   GID   INTERPRO_DESCRI… INTERPRO_E_VALUE INTERPRO_END_MIN INTERPRO_INTERP…
##   &lt;chr&gt; &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;           
## 1 Q0045 Cytochrome C an… 0.0              460              IPR000883       
## 2 Q0045 Cytochrome oxid… 110.0            534              IPR023616       
## 3 Q0045 Cytochrome c ox… 0.0              529              IPR036927       
## 4 Q0050 Reverse transcr… 5.6e-39          577              IPR000477       
## 5 Q0050 Type II intron … 1.4e-53          759              IPR024937       
## 6 Q0050 Reverse transcr… 46.0             577              IPR000477       
## # … with 5 more variables: INTERPRO_NAME &lt;chr&gt;, INTERPRO_PRIMARY_ID &lt;chr&gt;,
## #   INTERPRO_SECONDARY_ID &lt;chr&gt;, INTERPRO_START_MIN &lt;chr&gt;,
## #   INTERPRO_TRANSCRIPT_ID_S &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb134-1" title="1"><span class="co">## Interpro data for Crithidia!</span></a>
<a class="sourceLine" id="cb134-2" title="2"></a>
<a class="sourceLine" id="cb134-3" title="3">chosen_columns_idx &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="dt">x=</span>avail_columns, <span class="dt">pattern=</span><span class="st">&quot;^PATHWAY&quot;</span>)</a>
<a class="sourceLine" id="cb134-4" title="4">chosen_columns &lt;-<span class="st"> </span>avail_columns[chosen_columns_idx]</a>
<a class="sourceLine" id="cb134-5" title="5">sc_path &lt;-<span class="st"> </span><span class="kw">load_orgdb_go</span>(sc_orgdb, <span class="dt">columns=</span>chosen_columns)</a></code></pre></div>
<pre><code>## This is an orgdb, good.
## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb136-1" title="1"><span class="kw">head</span>(sc_path)</a></code></pre></div>
<pre><code>## # A tibble: 6 x 8
##   GID   PATHWAY_EC_NUMB… PATHWAY_EXACT_E… PATHWAY_EXPASY_… PATHWAY_PATHWAY
##   &lt;chr&gt; &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;            &lt;chr&gt;          
## 1 Q0045 1.9.3.1          Yes              http://enzyme.e… aerobic respir…
## 2 Q0045 1.9.3.1          Yes              http://enzyme.e… arsenite oxida…
## 3 Q0045 1.-.-.-          No               http://enzyme.e… sesamin biosyn…
## 4 Q0045 1.-.-.-          No               http://enzyme.e… hypoglycin bio…
## 5 Q0045 1.-.-.-          No               http://enzyme.e… 6-methoxypodop…
## 6 Q0045 1.-.-.-          No               http://enzyme.e… sorgoleone bio…
## # … with 3 more variables: PATHWAY_PATHWAY_SOURCE &lt;chr&gt;,
## #   PATHWAY_PATHWAY_SOURCE_ID &lt;chr&gt;,
## #   PATHWAY_REACTIONS_MATCHING_EC_NUMBER &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb138-1" title="1"><span class="co">## Pathway data for Crithidia!</span></a>
<a class="sourceLine" id="cb138-2" title="2"></a>
<a class="sourceLine" id="cb138-3" title="3"><span class="co">## This does not work for the moment because of some oddities with</span></a>
<a class="sourceLine" id="cb138-4" title="4"><span class="co">## the various tables at the eupathdb.  I have an email query with them</span></a>
<a class="sourceLine" id="cb138-5" title="5"><span class="co">## regarding it.</span></a>
<a class="sourceLine" id="cb138-6" title="6"><span class="co">##chosen_columns_idx &lt;- grepl(x=avail_columns, pattern=&quot;^ORTHOLOG&quot;)</span></a>
<a class="sourceLine" id="cb138-7" title="7"><span class="co">##chosen_columns &lt;- avail_columns[chosen_columns_idx]</span></a>
<a class="sourceLine" id="cb138-8" title="8"><span class="co">##chosen_columns &lt;- c(chosen_columns, &quot;ORGANISM&quot;)</span></a>
<a class="sourceLine" id="cb138-9" title="9"><span class="co">##crit_ortho &lt;- load_orgdb_go(sc_orgdb, columns=chosen_columns)</span></a>
<a class="sourceLine" id="cb138-10" title="10"><span class="co">##head(crit_ortho)</span></a>
<a class="sourceLine" id="cb138-11" title="11"><span class="co">## Orthologs!</span></a></code></pre></div>
</div>
<div id="shortcut-methods" class="section level3">
<h3>Shortcut methods</h3>
<p>Remembering the keytype and package names and everything can be annoying. The following attempts to make that easier. In this instance, the only thing we need to provide is a unique substring for the species of interest. If the substring is not unique, this should show the matches and choose the first.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb139-1" title="1"><span class="co">## The function load_eupath_annotations() provides a shortcut to the above.</span></a>
<a class="sourceLine" id="cb139-2" title="2">major_annot &lt;-<span class="st"> </span><span class="kw">load_eupath_annotations</span>(<span class="dt">species=</span><span class="st">&quot;major&quot;</span>)</a></code></pre></div>
<pre><code>## Added 34 species after using only the genus species.</code></pre>
<pre><code>## Found the following hits: Leishmania major strain Friedlin, Leishmania major strain LV39c5, Leishmania major strain SD 75.1, choosing the first.</code></pre>
<pre><code>## </code></pre>
<pre><code>## Returning fields which start with 'annot'</code></pre>
<pre><code>## Unable to find CDSNAME, setting it to ANNOT_GENE_NAME.</code></pre>
<pre><code>## Unable to find CDSCHROM in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSSTRAND in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSSTART in the db, removing it.</code></pre>
<pre><code>## Unable to find CDSEND in the db, removing it.</code></pre>
<pre><code>## Extracted all gene ids.</code></pre>
<pre><code>## Attempting to select: ANNOT_GENE_NAME, GENE_TYPE, GID, ANNOT_BFD3_CDS, ANNOT_BFD3_MODEL, ANNOT_BFD6_CDS, ANNOT_BFD6_MODEL, ANNOT_CDS, ANNOT_CDS_LENGTH, ANNOT_CHROMOSOME, ANNOT_DIF_CDS, ANNOT_DIF_MODEL, ANNOT_EC_NUMBERS, ANNOT_EC_NUMBERS_DERIVED, ANNOT_EXON_COUNT, ANNOT_FC_BFD3_CDS, ANNOT_FC_BFD3_MODEL, ANNOT_FC_BFD6_CDS, ANNOT_FC_BFD6_MODEL, ANNOT_FC_DIF_CDS, ANNOT_FC_DIF_MODEL, ANNOT_FC_PF_CDS, ANNOT_FC_PF_MODEL, ANNOT_FIVE_PRIME_UTR_LENGTH, ANNOT_GENE_ENTREZ_ID, ANNOT_GENE_EXON_COUNT, ANNOT_GENE_HTS_NONCODING_SNPS, ANNOT_GENE_HTS_NONSYN_SYN_RATIO, ANNOT_GENE_HTS_NONSYNONYMOUS_SNPS, ANNOT_GENE_HTS_STOP_CODON_SNPS, ANNOT_GENE_HTS_SYNONYMOUS_SNPS, ANNOT_GENE_LOCATION_TEXT, ANNOT_GENE_NAME, ANNOT_GENE_ORTHOLOG_NUMBER, ANNOT_GENE_ORTHOMCL_NAME, ANNOT_GENE_PARALOG_NUMBER, ANNOT_GENE_PREVIOUS_IDS, ANNOT_GENE_PRODUCT, ANNOT_GENE_SOURCE_ID, ANNOT_GENE_TOTAL_HTS_SNPS, ANNOT_GENE_TRANSCRIPT_COUNT, ANNOT_GENE_TYPE, ANNOT_GO_COMPONENT, ANNOT_GO_FUNCTION, ANNOT_GO_ID_COMPONENT, ANNOT_GO_ID_FUNCTION, ANNOT_GO_ID_PROCESS, ANNOT_GO_PROCESS, ANNOT_HAS_MISSING_TRANSCRIPTS, ANNOT_INTERPRO_DESCRIPTION, ANNOT_INTERPRO_ID, ANNOT_IS_PSEUDO, ANNOT_ISOELECTRIC_POINT, ANNOT_LOCATION_TEXT, ANNOT_MATCHED_RESULT, ANNOT_MOLECULAR_WEIGHT, ANNOT_NO_TET_CDS, ANNOT_NO_TET_MODEL, ANNOT_ORGANISM, ANNOT_PF_CDS, ANNOT_PF_MODEL, ANNOT_PFAM_DESCRIPTION, ANNOT_PFAM_ID, ANNOT_PIRSF_DESCRIPTION, ANNOT_PIRSF_ID, ANNOT_PREDICTED_GO_COMPONENT, ANNOT_PREDICTED_GO_FUNCTION, ANNOT_PREDICTED_GO_ID_COMPONENT, ANNOT_PREDICTED_GO_ID_FUNCTION, ANNOT_PREDICTED_GO_ID_PROCESS, ANNOT_PREDICTED_GO_PROCESS, ANNOT_PROJECT_ID, ANNOT_PROSITEPROFILES_DESCRIPTION, ANNOT_PROSITEPROFILES_ID, ANNOT_PROTEIN_LENGTH, ANNOT_PROTEIN_SEQUENCE, ANNOT_SEQUENCE_ID, ANNOT_SIGNALP_PEPTIDE, ANNOT_SIGNALP_SCORES, ANNOT_SMART_DESCRIPTION, ANNOT_SMART_ID, ANNOT_SOURCE_ID, ANNOT_STRAND, ANNOT_SUPERFAMILY_DESCRIPTION, ANNOT_SUPERFAMILY_ID, ANNOT_THREE_PRIME_UTR_LENGTH, ANNOT_TIGRFAM_DESCRIPTION, ANNOT_TIGRFAM_ID, ANNOT_TM_COUNT, ANNOT_TRANS_FOUND_PER_GENE_INTERNAL, ANNOT_TRANSCRIPT_INDEX_PER_GENE, ANNOT_TRANSCRIPT_LENGTH, ANNOT_TRANSCRIPT_LINK, ANNOT_TRANSCRIPT_PRODUCT, ANNOT_TRANSCRIPT_SEQUENCE, ANNOT_TRANSCRIPTS_FOUND_PER_GENE, ANNOT_UNIPROT_ID, ANNOT_URI, ANNOT_WDK_WEIGHT</code></pre>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb152-1" title="1"><span class="kw">dim</span>(major_annot)</a></code></pre></div>
<pre><code>## [1] 18238    97</code></pre>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb154-1" title="1"><span class="co">## This provides the same information as the results of the select up above.</span></a></code></pre></div>
</div>
<div id="shortcut-orthologs" class="section level3">
<h3>Shortcut orthologs</h3>
<p>A task I find myself needing to do fairly often is to get the set of genes orthologous to a given gene. There are lots of methods to handle this question; one of them includes the nice table provided by the eupathdb. Let us query that.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb155-1" title="1"><span class="co">## A recent EuPathDB update makes it possible to use the 'OrthologsLite' table rather than</span></a>
<a class="sourceLine" id="cb155-2" title="2"><span class="co">## Orthologs, which is much faster (by like 1000x), but I am quickly realizing much more</span></a>
<a class="sourceLine" id="cb155-3" title="3"><span class="co">## limited in the information it returns, and only exists for a subset of the eupathdb</span></a>
<a class="sourceLine" id="cb155-4" title="4"><span class="co">## projects.  As a result, I might just drop its usage and force the much slower queries</span></a>
<a class="sourceLine" id="cb155-5" title="5"><span class="co">## to the more complete table...</span></a>
<a class="sourceLine" id="cb155-6" title="6">major_entry &lt;-<span class="st"> </span><span class="kw">get_eupath_entry</span>(<span class="dt">species=</span><span class="st">&quot;major&quot;</span>, <span class="dt">webservice=</span><span class="st">&quot;tritrypdb&quot;</span>)</a>
<a class="sourceLine" id="cb155-7" title="7">major_pkg &lt;-<span class="st"> </span><span class="kw">get_eupath_pkgnames</span>(major_entry)</a>
<a class="sourceLine" id="cb155-8" title="8">major_orgdb &lt;-<span class="st"> </span>major_pkg<span class="op">$</span>orgdb</a>
<a class="sourceLine" id="cb155-9" title="9">lm_ortho &lt;-<span class="st"> </span><span class="kw">extract_eupath_orthologs</span>(major_orgdb)</a>
<a class="sourceLine" id="cb155-10" title="10"><span class="kw">dim</span>(lm_ortho)</a>
<a class="sourceLine" id="cb155-11" title="11"><span class="kw">head</span>(lm_ortho)</a>
<a class="sourceLine" id="cb155-12" title="12"><span class="kw">summary</span>(lm_ortho)</a></code></pre></div>
</div>
</div>
</div>
<div id="session-information" class="section level1">
<h1>Session Information</h1>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb156-1" title="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Devuan GNU/Linux beowulf/ceres
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.3.5.so
## 
## locale:
##  [1] LC_CTYPE=en_US.utf8       LC_NUMERIC=C             
##  [3] LC_TIME=en_US.utf8        LC_COLLATE=en_US.utf8    
##  [5] LC_MONETARY=en_US.utf8    LC_MESSAGES=en_US.utf8   
##  [7] LC_PAPER=en_US.utf8       LC_NAME=C                
##  [9] LC_ADDRESS=C              LC_TELEPHONE=C           
## [11] LC_MEASUREMENT=en_US.utf8 LC_IDENTIFICATION=C      
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Lmajor.Friedlin.v43.eg.db_2019.04
##  [2] org.Scerevisiae.v43.eg.db_2019.04    
##  [3] futile.logger_1.4.3                  
##  [4] AnnotationDbi_1.44.0                 
##  [5] EuPathDB_1.0.1                       
##  [6] AnnotationHub_2.14.5                 
##  [7] GenomeInfoDbData_1.2.0               
##  [8] GenomicRanges_1.34.0                 
##  [9] GenomeInfoDb_1.18.2                  
## [10] IRanges_2.16.0                       
## [11] S4Vectors_0.20.1                     
## [12] Biobase_2.42.0                       
## [13] BiocGenerics_0.28.0                  
## [14] BiocStyle_2.10.0                     
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6                  matrixStats_0.54.0           
##  [3] fs_1.2.7                      usethis_1.4.0                
##  [5] devtools_2.0.1                bit64_0.9-7                  
##  [7] progress_1.2.0                httr_1.4.0                   
##  [9] rprojroot_1.3-2               tools_3.6.0                  
## [11] backports_1.1.3               utf8_1.1.4                   
## [13] R6_2.4.0                      DBI_1.0.0                    
## [15] withr_2.1.2                   tidyselect_0.2.5             
## [17] prettyunits_1.0.2             processx_3.3.0               
## [19] bit_1.1-14                    curl_3.3                     
## [21] compiler_3.6.0                graph_1.60.0                 
## [23] cli_1.1.0                     rvest_0.3.2                  
## [25] formatR_1.6                   AnnotationHubData_1.12.0     
## [27] xml2_1.2.0                    desc_1.2.0                   
## [29] DelayedArray_0.8.0            rtracklayer_1.42.2           
## [31] RBGL_1.58.2                   callr_3.2.0                  
## [33] stringr_1.4.0                 digest_0.6.18                
## [35] Rsamtools_1.34.1              rmarkdown_1.12               
## [37] AnnotationForge_1.24.0        XVector_0.22.0               
## [39] rBiopaxParser_2.22.0          pkgconfig_2.0.2              
## [41] htmltools_0.3.6               sessioninfo_1.1.1            
## [43] rlang_0.3.3                   RSQLite_2.1.1                
## [45] BiocInstaller_1.32.1          shiny_1.2.0                  
## [47] jsonlite_1.6                  BiocParallel_1.16.6          
## [49] dplyr_0.8.0.1                 RCurl_1.95-4.12              
## [51] magrittr_1.5                  GO.db_3.7.0                  
## [53] Matrix_1.2-17                 fansi_0.4.0                  
## [55] Rcpp_1.0.1                    stringi_1.4.3                
## [57] yaml_2.2.0                    SummarizedExperiment_1.12.0  
## [59] zlibbioc_1.28.0               biocViews_1.50.10            
## [61] pkgbuild_1.0.3                grid_3.6.0                   
## [63] blob_1.1.1                    promises_1.0.1               
## [65] crayon_1.3.4                  lattice_0.20-38              
## [67] Biostrings_2.50.2             GenomicFeatures_1.34.7       
## [69] hms_0.4.2                     knitr_1.22                   
## [71] ps_1.3.0                      pillar_1.3.1                 
## [73] RUnit_0.4.32                  codetools_0.2-16             
## [75] biomaRt_2.38.0                pkgload_1.0.2                
## [77] futile.options_1.0.1          XML_3.98-1.19                
## [79] glue_1.3.1                    evaluate_0.13                
## [81] lambda.r_1.2.3                data.table_1.12.0            
## [83] remotes_2.0.2                 BiocManager_1.30.4           
## [85] httpuv_1.5.0                  foreach_1.4.4                
## [87] testthat_2.0.1                purrr_0.3.2                  
## [89] assertthat_0.2.1              xfun_0.6                     
## [91] mime_0.6                      xtable_1.8-3                 
## [93] later_0.8.0                   tibble_2.1.1                 
## [95] OrganismDbi_1.24.0            iterators_1.0.10             
## [97] GenomicAlignments_1.18.1      memoise_1.1.0                
## [99] interactiveDisplayBase_1.20.0</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
